<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Tree based analysis of single rna-seq clusters • sctree</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Tree based analysis of single rna-seq clusters">
<meta property="og:description" content="Attempts to streamline the translation of single-cell 
    rna-seq data to flow cytometry applications.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">sctree</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.8.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/antibody_query.html">Query Antibody vendors from inside R</a>
    </li>
    <li>
      <a href="articles/evaluating_tree_quality.html">Evaluating Tree Quality</a>
    </li>
    <li>
      <a href="articles/plot_flowstyle.html">Flow style plotting</a>
    </li>
    <li>
      <a href="articles/sctree/sctree.html">sctree</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div id="sctree-a-package-to-connect-single-cell-rna-seq-to-biology-using-trees" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#sctree-a-package-to-connect-single-cell-rna-seq-to-biology-using-trees" class="anchor"></a>sctree: a package to connect single cell rna-seq to biology using trees</h1></div>
<!-- README.md is generated from README.Rmd. Please edit that file -->

</div>
<div id="sctree" class="section level1">
<h1 class="hasAnchor">
<a href="#sctree" class="anchor"></a>sctree</h1>
<p>The goal of sctree is to create a tool to accelerate the transition from single cell rna-sequencing to calidation and new sub-population discovery.</p>
<p>Features suggesting pseudo-gating strategies to purify found populations via flow-cytometry, antibody querying and cross validations between datasets.</p>
<p>Please check our companion documentation website hosted at <a href="https://jspaezp.github.io/sctree/">jspaezp.github.io/sctree</a>.</p>
<p>Number of lines in roxygen comments: 1039</p>
<p>Number of lines in R code: 1968</p>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>Currently we are not hosted in CRAN but the package can be installed by using the following function in R</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" title="1">remotes<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/remotes/man/install_github.html">install_github</a></span>(<span class="st">"jspaezp/sctree"</span>)</a></code></pre></div>
</div>
<div id="usage" class="section level1">
<h1 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h1>
<p>I am assuming you have already done your clustering and dimensional reduction using seurat and we have our seurat object.</p>
<p>For this examples we will use a dummy dataset that come bundled with the package</p>
<p><code>small_5050_mix</code>, this dataset comes originally from the 1:1 mixture of Jurkat and 293T cells provided by 10x.</p>
<p>Original data can be found here:</p>
<ol>
<li><a href="https://support.10xgenomics.com/single-cell-gene-expression/datasets/1.1.0/jurkat:293t_50:50">1:1 mixture</a></li>
<li><a href="https://support.10xgenomics.com/single-cell-gene-expression/datasets/1.1.0/jurkat_293t_99_1">99:1 mixture</a></li>
</ol>
<!-- end list --><div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" title="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(sctree)</a>
<a class="sourceLine" id="cb2-2" title="2"><span class="co">#&gt; Loading required package: Seurat</span></a>
<a class="sourceLine" id="cb2-3" title="3"><span class="co">#&gt; Registered S3 method overwritten by 'GGally':</span></a>
<a class="sourceLine" id="cb2-4" title="4"><span class="co">#&gt;   method from   </span></a>
<a class="sourceLine" id="cb2-5" title="5"><span class="co">#&gt;   +.gg   ggplot2</span></a>
<a class="sourceLine" id="cb2-6" title="6"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-7" title="7"><span class="co">#&gt; Attaching package: 'sctree'</span></a>
<a class="sourceLine" id="cb2-8" title="8"><span class="co">#&gt; The following objects are masked from 'package:Seurat':</span></a>
<a class="sourceLine" id="cb2-9" title="9"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-10" title="10"><span class="co">#&gt;     FindAllMarkers, FindConservedMarkers, FindMarkers</span></a>
<a class="sourceLine" id="cb2-11" title="11"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(Seurat)</a>
<a class="sourceLine" id="cb2-12" title="12"></a>
<a class="sourceLine" id="cb2-13" title="13"><span class="kw"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="dv">6</span>)</a>
<a class="sourceLine" id="cb2-14" title="14"></a>
<a class="sourceLine" id="cb2-15" title="15"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(small_<span class="dv">5050</span>_mix)</a>
<a class="sourceLine" id="cb2-16" title="16">small_<span class="dv">5050</span>_mix</a>
<a class="sourceLine" id="cb2-17" title="17"><span class="co">#&gt; An object of class Seurat </span></a>
<a class="sourceLine" id="cb2-18" title="18"><span class="co">#&gt; 1031 features across 255 samples within 1 assay </span></a>
<a class="sourceLine" id="cb2-19" title="19"><span class="co">#&gt; Active assay: RNA (1031 features)</span></a>
<a class="sourceLine" id="cb2-20" title="20"><span class="co">#&gt;  2 dimensional reductions calculated: pca, tsne</span></a>
<a class="sourceLine" id="cb2-21" title="21"></a>
<a class="sourceLine" id="cb2-22" title="22"><span class="kw"><a href="https://rdrr.io/pkg/Seurat/man/DimPlot.html">DimPlot</a></span>(small_<span class="dv">5050</span>_mix, <span class="dt">reduction =</span> <span class="st">"tsne"</span>)</a></code></pre></div>
<p><img src="reference/figures/unnamed-chunk-8-1.png"><!-- --></p>
<div id="finding-important-variables-to-classify-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#finding-important-variables-to-classify-clusters" class="anchor"></a>Finding important variables to classify clusters</h2>
<p>We base our importances on the “classification value” they give to a random forest (using the implementation in the <code>ranger</code> package)</p>
<p>So lets fit the random forest … Here we are adding the <code>warn.imp.method</code> to prevent a warning message sent by <code>ranger</code> when most of the variables are correlated with the clustering.</p>
<p>Please reffer to the of the <code>importance_pvalues</code> section in the <a href="https://cran.r-project.org/web/packages/ranger/ranger.pdf">ranger documentation</a> when addressing this issue and for more details.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" title="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(sctree)</a>
<a class="sourceLine" id="cb3-2" title="2">rang_importances &lt;-<span class="st"> </span><span class="kw"><a href="reference/ranger_importances.html">ranger_importances</a></span>(</a>
<a class="sourceLine" id="cb3-3" title="3">    small_<span class="dv">5050</span>_mix,</a>
<a class="sourceLine" id="cb3-4" title="4">    <span class="dt">cluster =</span> <span class="st">"ALL"</span>,</a>
<a class="sourceLine" id="cb3-5" title="5">    <span class="dt">warn.imp.method =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<p>By default, we obtain a data frame containing only importances with pvalues under <code>0.05</code>.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" title="1"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(rang_importances)</a>
<a class="sourceLine" id="cb4-2" title="2"><span class="co">#&gt;         importance pvalue    gene</span></a>
<a class="sourceLine" id="cb4-3" title="3"><span class="co">#&gt; ASNS      5.636836      0    ASNS</span></a>
<a class="sourceLine" id="cb4-4" title="4"><span class="co">#&gt; ARHGDIB   4.597722      0 ARHGDIB</span></a>
<a class="sourceLine" id="cb4-5" title="5"><span class="co">#&gt; CD3D      4.466280      0    CD3D</span></a>
<a class="sourceLine" id="cb4-6" title="6"><span class="co">#&gt; TMSB4X    4.450082      0  TMSB4X</span></a>
<a class="sourceLine" id="cb4-7" title="7"><span class="co">#&gt; ADA       3.696396      0     ADA</span></a>
<a class="sourceLine" id="cb4-8" title="8"><span class="co">#&gt; MZB1      3.197148      0    MZB1</span></a></code></pre></div>
</div>
</div>
<div id="seurat-interface" class="section level1">
<h1 class="hasAnchor">
<a href="#seurat-interface" class="anchor"></a>Seurat Interface</h1>
<p>As an analogous function to Seurat’s <code>FindAllMarkers</code>, we offer <code>ranger_importances</code> or the <code>RangerDE</code> option for <code>FindAllMarkers</code></p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" title="1">markers &lt;-<span class="st"> </span><span class="kw"><a href="reference/FindAllMarkers.html">FindAllMarkers</a></span>(</a>
<a class="sourceLine" id="cb5-2" title="2">  small_<span class="dv">5050</span>_mix,</a>
<a class="sourceLine" id="cb5-3" title="3">  <span class="dt">warn.imp.method =</span> <span class="ot">FALSE</span>, </a>
<a class="sourceLine" id="cb5-4" title="4">  <span class="dt">test.use =</span> <span class="st">"RangerDE"</span>)</a>
<a class="sourceLine" id="cb5-5" title="5"><span class="co">#&gt; Calculating cluster 0</span></a>
<a class="sourceLine" id="cb5-6" title="6"><span class="co">#&gt; Calculating cluster 1</span></a>
<a class="sourceLine" id="cb5-7" title="7"></a>
<a class="sourceLine" id="cb5-8" title="8"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(markers)</a>
<a class="sourceLine" id="cb5-9" title="9"><span class="co">#&gt;         importance p_val    gene avg_logFC pct.1 pct.2 p_val_adj cluster</span></a>
<a class="sourceLine" id="cb5-10" title="10"><span class="co">#&gt; MZB1     0.9641974     0    MZB1  2.004299 0.828 0.227         0       0</span></a>
<a class="sourceLine" id="cb5-11" title="11"><span class="co">#&gt; CD3G     1.2004702     0    CD3G  1.849551 0.744 0.107         0       0</span></a>
<a class="sourceLine" id="cb5-12" title="12"><span class="co">#&gt; CD3D     3.2936540     0    CD3D  1.842246 0.872 0.373         0       0</span></a>
<a class="sourceLine" id="cb5-13" title="13"><span class="co">#&gt; ARHGDIB  2.3889365     0 ARHGDIB  1.787994 0.856 0.320         0       0</span></a>
<a class="sourceLine" id="cb5-14" title="14"><span class="co">#&gt; AIF1     0.4203789     0    AIF1  1.759311 0.756 0.173         0       0</span></a>
<a class="sourceLine" id="cb5-15" title="15"><span class="co">#&gt; FYB      0.8291528     0     FYB  1.750418 0.739 0.107         0       0</span></a>
<a class="sourceLine" id="cb5-16" title="16"></a>
<a class="sourceLine" id="cb5-17" title="17">plot.markers &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span>(rbind, <span class="kw"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/split.html">split</a></span>(markers, markers<span class="op">$</span>cluster), head, <span class="dv">3</span>))</a>
<a class="sourceLine" id="cb5-18" title="18"></a>
<a class="sourceLine" id="cb5-19" title="19"><span class="kw"><a href="https://rdrr.io/pkg/Seurat/man/FeaturePlot.html">FeaturePlot</a></span>(small_<span class="dv">5050</span>_mix, <span class="kw"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(plot.markers<span class="op">$</span>gene))</a></code></pre></div>
<p><img src="reference/figures/unnamed-chunk-11-1.png"><!-- --></p>
<p>Note how variable importances can be high if a marker is either preferentially present of preferentially absent. Therefore as a pre-filtering step we implemented a modified version of seurat’s “FindMarkers”</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" title="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(Seurat)</a>
<a class="sourceLine" id="cb6-2" title="2"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(sctree)</a>
<a class="sourceLine" id="cb6-3" title="3"></a>
<a class="sourceLine" id="cb6-4" title="4"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(</a>
<a class="sourceLine" id="cb6-5" title="5">    sctree<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/sctree/man/FindMarkers.html">FindMarkers</a></span>(</a>
<a class="sourceLine" id="cb6-6" title="6">        small_<span class="dv">5050</span>_mix,</a>
<a class="sourceLine" id="cb6-7" title="7">        <span class="dt">features =</span> <span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(small_<span class="dv">5050</span>_mix<span class="op">@</span>assays<span class="op">$</span>RNA<span class="op">@</span>data),</a>
<a class="sourceLine" id="cb6-8" title="8">        <span class="dt">ident.1 =</span> <span class="dv">0</span>, <span class="dt">test.use =</span> <span class="st">"RangerDE"</span>))</a>
<a class="sourceLine" id="cb6-9" title="9"><span class="co">#&gt;         importance p_val    gene avg_logFC pct.1 pct.2 p_val_adj</span></a>
<a class="sourceLine" id="cb6-10" title="10"><span class="co">#&gt; MZB1     0.9641974     0    MZB1  2.004299 0.828 0.227         0</span></a>
<a class="sourceLine" id="cb6-11" title="11"><span class="co">#&gt; CD3G     1.2004702     0    CD3G  1.849551 0.744 0.107         0</span></a>
<a class="sourceLine" id="cb6-12" title="12"><span class="co">#&gt; CD3D     3.2936540     0    CD3D  1.842246 0.872 0.373         0</span></a>
<a class="sourceLine" id="cb6-13" title="13"><span class="co">#&gt; ARHGDIB  2.3889365     0 ARHGDIB  1.787994 0.856 0.320         0</span></a>
<a class="sourceLine" id="cb6-14" title="14"><span class="co">#&gt; AIF1     0.4203789     0    AIF1  1.759311 0.756 0.173         0</span></a>
<a class="sourceLine" id="cb6-15" title="15"><span class="co">#&gt; FYB      0.8291528     0     FYB  1.750418 0.739 0.107         0</span></a></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" title="1"></a>
<a class="sourceLine" id="cb7-2" title="2">markers &lt;-<span class="st"> </span>sctree<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/sctree/man/FindAllMarkers.html">FindAllMarkers</a></span>(</a>
<a class="sourceLine" id="cb7-3" title="3">        small_<span class="dv">5050</span>_mix,</a>
<a class="sourceLine" id="cb7-4" title="4">        <span class="dt">features =</span> <span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(small_<span class="dv">5050</span>_mix<span class="op">@</span>assays<span class="op">$</span>RNA<span class="op">@</span>data),</a>
<a class="sourceLine" id="cb7-5" title="5">        <span class="dt">test.use =</span> <span class="st">"RangerDE"</span>)</a>
<a class="sourceLine" id="cb7-6" title="6"><span class="co">#&gt; Calculating cluster 0</span></a>
<a class="sourceLine" id="cb7-7" title="7"><span class="co">#&gt; Calculating cluster 1</span></a>
<a class="sourceLine" id="cb7-8" title="8"></a>
<a class="sourceLine" id="cb7-9" title="9"><span class="co"># Here we just extract the top 3 markers for each cluster</span></a>
<a class="sourceLine" id="cb7-10" title="10">plot.markers &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span>(rbind, <span class="kw"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/split.html">split</a></span>(markers, markers<span class="op">$</span>cluster), head, <span class="dv">3</span>))</a>
<a class="sourceLine" id="cb7-11" title="11"></a>
<a class="sourceLine" id="cb7-12" title="12"></a>
<a class="sourceLine" id="cb7-13" title="13">plot.markers</a>
<a class="sourceLine" id="cb7-14" title="14"><span class="co">#&gt;          importance p_val gene avg_logFC pct.1 pct.2 p_val_adj cluster</span></a>
<a class="sourceLine" id="cb7-15" title="15"><span class="co">#&gt; 0.MZB1    0.9641974     0 MZB1  2.004299 0.828 0.227         0       0</span></a>
<a class="sourceLine" id="cb7-16" title="16"><span class="co">#&gt; 0.CD3G    1.2004702     0 CD3G  1.849551 0.744 0.107         0       0</span></a>
<a class="sourceLine" id="cb7-17" title="17"><span class="co">#&gt; 0.CD3D    3.2936540     0 CD3D  1.842246 0.872 0.373         0       0</span></a>
<a class="sourceLine" id="cb7-18" title="18"><span class="co">#&gt; 1.ASNS.1  3.7143112     0 ASNS  1.428501 0.787 0.217         0       1</span></a>
<a class="sourceLine" id="cb7-19" title="19"><span class="co">#&gt; 1.CA2.1   0.4797642     0  CA2  1.297591 0.933 0.494         0       1</span></a>
<a class="sourceLine" id="cb7-20" title="20"><span class="co">#&gt; 1.HEY1.1  1.1533167     0 HEY1  1.295811 0.893 0.389         0       1</span></a></code></pre></div>
<div id="visualizing-the-expected-outcome-of-a-flow-cytometry-experiment" class="section level2">
<h2 class="hasAnchor">
<a href="#visualizing-the-expected-outcome-of-a-flow-cytometry-experiment" class="anchor"></a>Visualizing the expected outcome of a flow cytometry experiment</h2>
<p>Lets say we choose the top n markers from the former list and we did a flow experiment … HYPOTHETICALLY the marker distribution would resemble the rna expression profile for which we have the function <code>plot_flowstyle</code></p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" title="1">top_markers</a>
<a class="sourceLine" id="cb8-2" title="2"><span class="co">#&gt; [1] "MZB1" "CD3G" "CD3D" "ASNS" "CA2"  "HEY1"</span></a></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" title="1">g &lt;-<span class="st"> </span><span class="kw"><a href="reference/plot_flowstyle.html">plot_flowstyle</a></span>(small_<span class="dv">5050</span>_mix, <span class="dt">markernames =</span> top_markers)</a>
<a class="sourceLine" id="cb9-2" title="2">g</a></code></pre></div>
<p><img src="reference/figures/unnamed-chunk-16-1.png"><!-- --></p>
<p>We can also focus in one of the pannels (and check the color conventions)</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" title="1">g[<span class="dv">1</span>,<span class="dv">2</span>]</a></code></pre></div>
<p><img src="reference/figures/unnamed-chunk-17-1.png"><!-- --></p>
</div>
<div id="suggesting-a-gating-strategy-for-the-markers" class="section level2">
<h2 class="hasAnchor">
<a href="#suggesting-a-gating-strategy-for-the-markers" class="anchor"></a>Suggesting a gating strategy for the markers</h2>
<p>A general strategy to get separate all clusters</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" title="1">top_markers</a>
<a class="sourceLine" id="cb11-2" title="2"><span class="co">#&gt; [1] "MZB1" "CD3G" "CD3D" "ASNS" "CA2"  "HEY1"</span></a></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" title="1">tree_fit &lt;-<span class="st"> </span><span class="kw"><a href="reference/fit_ctree.html">fit_ctree</a></span>(small_<span class="dv">5050</span>_mix,</a>
<a class="sourceLine" id="cb12-2" title="2">                      <span class="dt">genes_use =</span> top_markers,</a>
<a class="sourceLine" id="cb12-3" title="3">                      <span class="dt">cluster =</span> <span class="st">"ALL"</span>)</a></code></pre></div>
<p>Visualizing the tree as … a tree … we can see how our model is a simple series of yes/no questions.</p>
<p>If we wanted to classifiy a random cell: in the first <code>node</code>, we check if the expression of that gene is higher or lower than a given value, if it is lower, we proceed to the left, if not we go right. We keep doing that until we have no more <code>branches</code>. This final node will have a predicted cluster, in this plot we can also see how pure can we expect this group to be and how many of the cells in our training set clasify as part of it.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" title="1"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(tree_fit)</a></code></pre></div>
<p><img src="reference/figures/unnamed-chunk-21-1.png"><!-- --></p>
<p>When inspecting the tree_fit, we can see a more detailed text representation of this tree.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" title="1"><span class="kw"><a href="https://rdrr.io/pkg/Seurat/man/print.DimReduc.html">print</a></span>(tree_fit)</a>
<a class="sourceLine" id="cb14-2" title="2"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb14-3" title="3"><span class="co">#&gt; Model formula:</span></a>
<a class="sourceLine" id="cb14-4" title="4"><span class="co">#&gt; ident ~ MZB1 + CD3G + CD3D + ASNS + CA2 + HEY1</span></a>
<a class="sourceLine" id="cb14-5" title="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb14-6" title="6"><span class="co">#&gt; Fitted party:</span></a>
<a class="sourceLine" id="cb14-7" title="7"><span class="co">#&gt; [1] root</span></a>
<a class="sourceLine" id="cb14-8" title="8"><span class="co">#&gt; |   [2] CD3D &lt;= 2.30877</span></a>
<a class="sourceLine" id="cb14-9" title="9"><span class="co">#&gt; |   |   [3] ASNS &lt;= 1.43241: 0 (n = 37, err = 37.8%)</span></a>
<a class="sourceLine" id="cb14-10" title="10"><span class="co">#&gt; |   |   [4] ASNS &gt; 1.43241: 1 (n = 61, err = 16.4%)</span></a>
<a class="sourceLine" id="cb14-11" title="11"><span class="co">#&gt; |   [5] CD3D &gt; 2.30877</span></a>
<a class="sourceLine" id="cb14-12" title="12"><span class="co">#&gt; |   |   [6] ASNS &lt;= 2.10406: 0 (n = 150, err = 3.3%)</span></a>
<a class="sourceLine" id="cb14-13" title="13"><span class="co">#&gt; |   |   [7] ASNS &gt; 2.10406: 1 (n = 7, err = 28.6%)</span></a>
<a class="sourceLine" id="cb14-14" title="14"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb14-15" title="15"><span class="co">#&gt; Number of inner nodes:    3</span></a>
<a class="sourceLine" id="cb14-16" title="16"><span class="co">#&gt; Number of terminal nodes: 4</span></a></code></pre></div>
<p>Sometimes one might think that the proposed strategy is too complicated or not implementable in the experimental settings, in order to add constraints to the fit one can give additional arguments that will be passed to <code><a href="https://rdrr.io/pkg/partykit/man/ctree_control.html">partykit::ctree_control</a></code>, such as <code>maxdepth = 2</code> (maximum 2 questions per cell)</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" title="1">tree_fit &lt;-<span class="st"> </span><span class="kw"><a href="reference/fit_ctree.html">fit_ctree</a></span>(</a>
<a class="sourceLine" id="cb15-2" title="2">    small_<span class="dv">5050</span>_mix, <span class="dt">genes_use =</span> top_markers,</a>
<a class="sourceLine" id="cb15-3" title="3">    <span class="dt">cluster =</span> <span class="st">"ALL"</span>, <span class="dt">maxdepth =</span> <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb15-4" title="4"><span class="kw"><a href="https://rdrr.io/pkg/Seurat/man/print.DimReduc.html">print</a></span>(tree_fit)</a>
<a class="sourceLine" id="cb15-5" title="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb15-6" title="6"><span class="co">#&gt; Model formula:</span></a>
<a class="sourceLine" id="cb15-7" title="7"><span class="co">#&gt; ident ~ MZB1 + CD3G + CD3D + ASNS + CA2 + HEY1</span></a>
<a class="sourceLine" id="cb15-8" title="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb15-9" title="9"><span class="co">#&gt; Fitted party:</span></a>
<a class="sourceLine" id="cb15-10" title="10"><span class="co">#&gt; [1] root</span></a>
<a class="sourceLine" id="cb15-11" title="11"><span class="co">#&gt; |   [2] CD3D &lt;= 2.30877</span></a>
<a class="sourceLine" id="cb15-12" title="12"><span class="co">#&gt; |   |   [3] ASNS &lt;= 1.43241: 0 (n = 37, err = 37.8%)</span></a>
<a class="sourceLine" id="cb15-13" title="13"><span class="co">#&gt; |   |   [4] ASNS &gt; 1.43241: 1 (n = 61, err = 16.4%)</span></a>
<a class="sourceLine" id="cb15-14" title="14"><span class="co">#&gt; |   [5] CD3D &gt; 2.30877</span></a>
<a class="sourceLine" id="cb15-15" title="15"><span class="co">#&gt; |   |   [6] ASNS &lt;= 2.10406: 0 (n = 150, err = 3.3%)</span></a>
<a class="sourceLine" id="cb15-16" title="16"><span class="co">#&gt; |   |   [7] ASNS &gt; 2.10406: 1 (n = 7, err = 28.6%)</span></a>
<a class="sourceLine" id="cb15-17" title="17"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb15-18" title="18"><span class="co">#&gt; Number of inner nodes:    3</span></a>
<a class="sourceLine" id="cb15-19" title="19"><span class="co">#&gt; Number of terminal nodes: 4</span></a>
<a class="sourceLine" id="cb15-20" title="20"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(tree_fit)</a></code></pre></div>
<p><img src="reference/figures/unnamed-chunk-23-1.png"><!-- --></p>
<p>Since not all variables are ultimately used in our classifier, one can access the ones that were by using <code>varimp(tree_fit)</code></p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" title="1">partykit<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/partykit/man/varimp.html">varimp</a></span>(tree_fit)</a>
<a class="sourceLine" id="cb16-2" title="2"><span class="co">#&gt;      CD3D      ASNS </span></a>
<a class="sourceLine" id="cb16-3" title="3"><span class="co">#&gt; 0.2849840 0.1505534</span></a>
<a class="sourceLine" id="cb16-4" title="4"><span class="kw"><a href="reference/plot_flowstyle.html">plot_flowstyle</a></span>(small_<span class="dv">5050</span>_mix, <span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(partykit<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/partykit/man/varimp.html">varimp</a></span>(tree_fit)))</a></code></pre></div>
<p><img src="reference/figures/unnamed-chunk-24-1.png"><!-- --></p>
<p>One can also request the package to suggest a specific strategy only for a given cluster. This function is not expected to give drastically different results in datasets with few clusters, but it can definitely come usefull when many clusters are present and one is interested in a specific one.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" title="1">tree_fit &lt;-<span class="st"> </span><span class="kw"><a href="reference/fit_ctree.html">fit_ctree</a></span>(small_<span class="dv">5050</span>_mix, <span class="dt">genes_use =</span> top_markers, <span class="dt">cluster =</span> <span class="st">"0"</span>)</a>
<a class="sourceLine" id="cb17-2" title="2"><span class="kw"><a href="https://rdrr.io/pkg/Seurat/man/print.DimReduc.html">print</a></span>(tree_fit)</a>
<a class="sourceLine" id="cb17-3" title="3"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb17-4" title="4"><span class="co">#&gt; Model formula:</span></a>
<a class="sourceLine" id="cb17-5" title="5"><span class="co">#&gt; ident ~ MZB1 + CD3G + CD3D + ASNS + CA2 + HEY1</span></a>
<a class="sourceLine" id="cb17-6" title="6"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb17-7" title="7"><span class="co">#&gt; Fitted party:</span></a>
<a class="sourceLine" id="cb17-8" title="8"><span class="co">#&gt; [1] root</span></a>
<a class="sourceLine" id="cb17-9" title="9"><span class="co">#&gt; |   [2] CD3D &lt;= 2.30877</span></a>
<a class="sourceLine" id="cb17-10" title="10"><span class="co">#&gt; |   |   [3] ASNS &lt;= 1.43241: clus 0 (n = 37, err = 37.8%)</span></a>
<a class="sourceLine" id="cb17-11" title="11"><span class="co">#&gt; |   |   [4] ASNS &gt; 1.43241: not clus 0 (n = 61, err = 16.4%)</span></a>
<a class="sourceLine" id="cb17-12" title="12"><span class="co">#&gt; |   [5] CD3D &gt; 2.30877</span></a>
<a class="sourceLine" id="cb17-13" title="13"><span class="co">#&gt; |   |   [6] ASNS &lt;= 2.10406: clus 0 (n = 150, err = 3.3%)</span></a>
<a class="sourceLine" id="cb17-14" title="14"><span class="co">#&gt; |   |   [7] ASNS &gt; 2.10406: not clus 0 (n = 7, err = 28.6%)</span></a>
<a class="sourceLine" id="cb17-15" title="15"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb17-16" title="16"><span class="co">#&gt; Number of inner nodes:    3</span></a>
<a class="sourceLine" id="cb17-17" title="17"><span class="co">#&gt; Number of terminal nodes: 4</span></a></code></pre></div>
<p>Sometimes it is useful to visualize directly the subset of cells that will be “gated” out in each rule, this can be easily achieved by using our implementation of <code>plot_gates</code></p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" title="1"><span class="kw"><a href="reference/plot_gates.html">plot_gates</a></span>(small_<span class="dv">5050</span>_mix, tree_fit, <span class="st">"6"</span>)</a></code></pre></div>
<p><img src="reference/figures/unnamed-chunk-26-1.png"><!-- --></p>
<p>We have also implemented a way to export these rules as a <code>garnett</code> classifier. for more detail on how the classifier is implemented please refer to <a href="https://cole-trapnell-lab.github.io/garnett/docs/#constructing-a-marker-file">the garnett documentation</a></p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" title="1"><span class="kw"><a href="reference/as.garnett.html">as.garnett</a></span>(tree_fit)</a>
<a class="sourceLine" id="cb19-2" title="2"><span class="co">#&gt; &gt; clus 0_node_3  (n = 37)</span></a>
<a class="sourceLine" id="cb19-3" title="3"><span class="co">#&gt; expressed below: ASNS 1.432, CD3D 2.309</span></a>
<a class="sourceLine" id="cb19-4" title="4"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb19-5" title="5"><span class="co">#&gt; &gt; clus 0_node_6  (n = 150)</span></a>
<a class="sourceLine" id="cb19-6" title="6"><span class="co">#&gt; expressed above: CD3D 2.309</span></a>
<a class="sourceLine" id="cb19-7" title="7"><span class="co">#&gt; expressed below: ASNS 2.104</span></a>
<a class="sourceLine" id="cb19-8" title="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb19-9" title="9"><span class="co">#&gt; &gt; not clus 0_node_4  (n = 61)</span></a>
<a class="sourceLine" id="cb19-10" title="10"><span class="co">#&gt; expressed above: ASNS 1.432</span></a>
<a class="sourceLine" id="cb19-11" title="11"><span class="co">#&gt; expressed below: CD3D 2.309</span></a>
<a class="sourceLine" id="cb19-12" title="12"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb19-13" title="13"><span class="co">#&gt; &gt; not clus 0_node_7  (n = 7)</span></a>
<a class="sourceLine" id="cb19-14" title="14"><span class="co">#&gt; expressed above: ASNS 2.104, CD3D 2.309</span></a></code></pre></div>
</div>
<div id="finding-antibodies-for-the-experiment" class="section level2">
<h2 class="hasAnchor">
<a href="#finding-antibodies-for-the-experiment" class="anchor"></a>Finding antibodies for the experiment</h2>
<p>Since we acknowledge most experimental workflows need antibodies. We have implemented several functions to look for antibodies in vendor websites, as well as some helper functions to find the other posible aliases a gene might have.</p>
<p>Here is a simple example for a gene widely know to have an antibody available</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" title="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(sctree)</a>
<a class="sourceLine" id="cb20-2" title="2"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw"><a href="reference/antibodies.html">query_biocompare_antibodies</a></span>(<span class="st">"CD11b"</span>))</a>
<a class="sourceLine" id="cb20-3" title="3"><span class="co">#&gt;                                                                   title</span></a>
<a class="sourceLine" id="cb20-4" title="4"><span class="co">#&gt; 1 Anti-CD11b (integrin alpha-M) Rabbit Monoclonal Antibody, Clone#RM290</span></a>
<a class="sourceLine" id="cb20-5" title="5"><span class="co">#&gt; 2                                 Anti-CD11b Rabbit Monoclonal Antibody</span></a>
<a class="sourceLine" id="cb20-6" title="6"><span class="co">#&gt; 3                                    Anti-CD11b/ITGAM Picoband Antibody</span></a>
<a class="sourceLine" id="cb20-7" title="7"><span class="co">#&gt; 4                      Anti-Human CD11b DyLight 488 conjugated Antibody</span></a>
<a class="sourceLine" id="cb20-8" title="8"><span class="co">#&gt; 5                                  Monoclonal Antibody to CD11b (human)</span></a>
<a class="sourceLine" id="cb20-9" title="9"><span class="co">#&gt; 6                                                 CD11b antibody (FITC)</span></a>
<a class="sourceLine" id="cb20-10" title="10"><span class="co">#&gt;            vendor</span></a>
<a class="sourceLine" id="cb20-11" title="11"><span class="co">#&gt; 1       BosterBio</span></a>
<a class="sourceLine" id="cb20-12" title="12"><span class="co">#&gt; 2       BosterBio</span></a>
<a class="sourceLine" id="cb20-13" title="13"><span class="co">#&gt; 3       BosterBio</span></a>
<a class="sourceLine" id="cb20-14" title="14"><span class="co">#&gt; 4       BosterBio</span></a>
<a class="sourceLine" id="cb20-15" title="15"><span class="co">#&gt; 5 MyBioSource.com</span></a>
<a class="sourceLine" id="cb20-16" title="16"><span class="co">#&gt; 6 MyBioSource.com</span></a>
<a class="sourceLine" id="cb20-17" title="17"><span class="co">#&gt;                                                                                                       specification</span></a>
<a class="sourceLine" id="cb20-18" title="18"><span class="co">#&gt; 1                       Applications: WB, IHC; Reactivity: Human (Hu); Conjugate/Tag: Unconjugated; Quantity: 100uL</span></a>
<a class="sourceLine" id="cb20-19" title="19"><span class="co">#&gt; 2                  Applications: WB, IF, IHC; Reactivity: Hu, Ms; Conjugate/Tag: Unconjugated; Quantity: 100ug/vial</span></a>
<a class="sourceLine" id="cb20-20" title="20"><span class="co">#&gt; 3 Applications: WB, ELISA, FCM, ICC, IHC; Reactivity: Hu, Ms, Rt; Conjugate/Tag: Unconjugated; Quantity: 100ug/vial</span></a>
<a class="sourceLine" id="cb20-21" title="21"><span class="co">#&gt; 4      Applications: Flow Cytometry (FCM); Reactivity: Human (Hu); Conjugate/Tag: DyLight®488; Quantity: 100ug/vial</span></a>
<a class="sourceLine" id="cb20-22" title="22"><span class="co">#&gt; 5         Applications: Flow Cytometry (FCM); Reactivity: Human (Hu); Conjugate/Tag: Unconjugated; Quantity: 0.1 mg</span></a>
<a class="sourceLine" id="cb20-23" title="23"><span class="co">#&gt; 6                                                                     Conjugate/Tag: Unconjugated; Quantity: 0.5 mg</span></a></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" title="1"><span class="kw"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()</a>
<a class="sourceLine" id="cb21-2" title="2"><span class="co">#&gt; R version 3.6.2 (2019-12-12)</span></a>
<a class="sourceLine" id="cb21-3" title="3"><span class="co">#&gt; Platform: x86_64-redhat-linux-gnu (64-bit)</span></a>
<a class="sourceLine" id="cb21-4" title="4"><span class="co">#&gt; Running under: Fedora 30 (Workstation Edition)</span></a>
<a class="sourceLine" id="cb21-5" title="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb21-6" title="6"><span class="co">#&gt; Matrix products: default</span></a>
<a class="sourceLine" id="cb21-7" title="7"><span class="co">#&gt; BLAS/LAPACK: /usr/lib64/R/lib/libRblas.so</span></a>
<a class="sourceLine" id="cb21-8" title="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb21-9" title="9"><span class="co">#&gt; locale:</span></a>
<a class="sourceLine" id="cb21-10" title="10"><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></a>
<a class="sourceLine" id="cb21-11" title="11"><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></a>
<a class="sourceLine" id="cb21-12" title="12"><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></a>
<a class="sourceLine" id="cb21-13" title="13"><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></a>
<a class="sourceLine" id="cb21-14" title="14"><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></a>
<a class="sourceLine" id="cb21-15" title="15"><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></a>
<a class="sourceLine" id="cb21-16" title="16"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb21-17" title="17"><span class="co">#&gt; attached base packages:</span></a>
<a class="sourceLine" id="cb21-18" title="18"><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></a>
<a class="sourceLine" id="cb21-19" title="19"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb21-20" title="20"><span class="co">#&gt; other attached packages:</span></a>
<a class="sourceLine" id="cb21-21" title="21"><span class="co">#&gt; [1] sctree_0.0.7.0000 Seurat_3.1.1     </span></a>
<a class="sourceLine" id="cb21-22" title="22"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb21-23" title="23"><span class="co">#&gt; loaded via a namespace (and not attached):</span></a>
<a class="sourceLine" id="cb21-24" title="24"><span class="co">#&gt;   [1] Rtsne_0.15           colorspace_1.4-1     selectr_0.4-2       </span></a>
<a class="sourceLine" id="cb21-25" title="25"><span class="co">#&gt;   [4] ggridges_0.5.1       farver_2.0.1         leiden_0.3.1        </span></a>
<a class="sourceLine" id="cb21-26" title="26"><span class="co">#&gt;   [7] listenv_0.8.0        npsurv_0.4-0         ggrepel_0.8.1       </span></a>
<a class="sourceLine" id="cb21-27" title="27"><span class="co">#&gt;  [10] bit64_0.9-7          mvtnorm_1.0-11       AnnotationDbi_1.46.1</span></a>
<a class="sourceLine" id="cb21-28" title="28"><span class="co">#&gt;  [13] xml2_1.2.2           ranger_0.12.1        codetools_0.2-16    </span></a>
<a class="sourceLine" id="cb21-29" title="29"><span class="co">#&gt;  [16] splines_3.6.2        R.methodsS3_1.7.1    lsei_1.2-0          </span></a>
<a class="sourceLine" id="cb21-30" title="30"><span class="co">#&gt;  [19] libcoin_1.0-5        knitr_1.26           Formula_1.2-3       </span></a>
<a class="sourceLine" id="cb21-31" title="31"><span class="co">#&gt;  [22] jsonlite_1.6         ica_1.0-2            cluster_2.1.0       </span></a>
<a class="sourceLine" id="cb21-32" title="32"><span class="co">#&gt;  [25] png_0.1-7            R.oo_1.23.0          uwot_0.1.5          </span></a>
<a class="sourceLine" id="cb21-33" title="33"><span class="co">#&gt;  [28] sctransform_0.2.0    compiler_3.6.2       httr_1.4.1          </span></a>
<a class="sourceLine" id="cb21-34" title="34"><span class="co">#&gt;  [31] assertthat_0.2.1     Matrix_1.2-18        lazyeval_0.2.2      </span></a>
<a class="sourceLine" id="cb21-35" title="35"><span class="co">#&gt;  [34] htmltools_0.4.0      tools_3.6.2          rsvd_1.0.2          </span></a>
<a class="sourceLine" id="cb21-36" title="36"><span class="co">#&gt;  [37] igraph_1.2.4.2       partykit_1.2-5       gtable_0.3.0        </span></a>
<a class="sourceLine" id="cb21-37" title="37"><span class="co">#&gt;  [40] glue_1.3.1           RANN_2.6.1           reshape2_1.4.3      </span></a>
<a class="sourceLine" id="cb21-38" title="38"><span class="co">#&gt;  [43] dplyr_0.8.3          Rcpp_1.0.3           Biobase_2.44.0      </span></a>
<a class="sourceLine" id="cb21-39" title="39"><span class="co">#&gt;  [46] vctrs_0.2.2          gdata_2.18.0         ape_5.3             </span></a>
<a class="sourceLine" id="cb21-40" title="40"><span class="co">#&gt;  [49] nlme_3.1-142         gbRd_0.4-11          lmtest_0.9-37       </span></a>
<a class="sourceLine" id="cb21-41" title="41"><span class="co">#&gt;  [52] inum_1.0-1           xfun_0.11            stringr_1.4.0       </span></a>
<a class="sourceLine" id="cb21-42" title="42"><span class="co">#&gt;  [55] globals_0.12.5       rvest_0.3.5          lifecycle_0.1.0     </span></a>
<a class="sourceLine" id="cb21-43" title="43"><span class="co">#&gt;  [58] irlba_2.3.3          gtools_3.8.1         future_1.16.0       </span></a>
<a class="sourceLine" id="cb21-44" title="44"><span class="co">#&gt;  [61] MASS_7.3-51.4        zoo_1.8-6            scales_1.1.0        </span></a>
<a class="sourceLine" id="cb21-45" title="45"><span class="co">#&gt;  [64] parallel_3.6.2       RColorBrewer_1.1-2   yaml_2.2.0          </span></a>
<a class="sourceLine" id="cb21-46" title="46"><span class="co">#&gt;  [67] memoise_1.1.0        reticulate_1.13      pbapply_1.4-2       </span></a>
<a class="sourceLine" id="cb21-47" title="47"><span class="co">#&gt;  [70] gridExtra_2.3        ggplot2_3.2.1        rpart_4.1-15        </span></a>
<a class="sourceLine" id="cb21-48" title="48"><span class="co">#&gt;  [73] reshape_0.8.8        stringi_1.4.6        RSQLite_2.1.4       </span></a>
<a class="sourceLine" id="cb21-49" title="49"><span class="co">#&gt;  [76] S4Vectors_0.22.1     caTools_1.17.1.3     BiocGenerics_0.30.0 </span></a>
<a class="sourceLine" id="cb21-50" title="50"><span class="co">#&gt;  [79] bibtex_0.4.2         Rdpack_0.11-0        SDMTools_1.1-221.2  </span></a>
<a class="sourceLine" id="cb21-51" title="51"><span class="co">#&gt;  [82] rlang_0.4.4          pkgconfig_2.0.3      bitops_1.0-6        </span></a>
<a class="sourceLine" id="cb21-52" title="52"><span class="co">#&gt;  [85] evaluate_0.14        lattice_0.20-38      ROCR_1.0-7          </span></a>
<a class="sourceLine" id="cb21-53" title="53"><span class="co">#&gt;  [88] purrr_0.3.3          labeling_0.3         htmlwidgets_1.5.1   </span></a>
<a class="sourceLine" id="cb21-54" title="54"><span class="co">#&gt;  [91] cowplot_1.0.0        bit_1.1-14           tidyselect_0.2.5    </span></a>
<a class="sourceLine" id="cb21-55" title="55"><span class="co">#&gt;  [94] GGally_1.4.0         RcppAnnoy_0.0.14     wrapr_1.9.3         </span></a>
<a class="sourceLine" id="cb21-56" title="56"><span class="co">#&gt;  [97] plyr_1.8.5           magrittr_1.5         R6_2.4.1            </span></a>
<a class="sourceLine" id="cb21-57" title="57"><span class="co">#&gt; [100] IRanges_2.18.3       gplots_3.0.1.1       DBI_1.0.0           </span></a>
<a class="sourceLine" id="cb21-58" title="58"><span class="co">#&gt; [103] pillar_1.4.3         fitdistrplus_1.0-14  survival_3.1-8      </span></a>
<a class="sourceLine" id="cb21-59" title="59"><span class="co">#&gt; [106] tibble_2.1.3         future.apply_1.3.0   tsne_0.1-3          </span></a>
<a class="sourceLine" id="cb21-60" title="60"><span class="co">#&gt; [109] crayon_1.3.4         KernSmooth_2.23-16   plotly_4.9.1        </span></a>
<a class="sourceLine" id="cb21-61" title="61"><span class="co">#&gt; [112] rmarkdown_1.18       viridis_0.5.1        grid_3.6.2          </span></a>
<a class="sourceLine" id="cb21-62" title="62"><span class="co">#&gt; [115] data.table_1.12.6    blob_1.2.0           metap_1.1           </span></a>
<a class="sourceLine" id="cb21-63" title="63"><span class="co">#&gt; [118] digest_0.6.24        tidyr_1.0.0          R.utils_2.9.1       </span></a>
<a class="sourceLine" id="cb21-64" title="64"><span class="co">#&gt; [121] RcppParallel_4.4.4   stats4_3.6.2         munsell_0.5.0       </span></a>
<a class="sourceLine" id="cb21-65" title="65"><span class="co">#&gt; [124] viridisLite_0.3.0</span></a></code></pre></div>
</div>
</div>
<div id="steps-down-the-road" class="section level1">
<h1 class="hasAnchor">
<a href="#steps-down-the-road" class="anchor"></a>Steps down the road</h1>
<ol>
<li>Address some of the TODO’s in this repository</li>
<li>Reduce dependecies by replacing functions to base equivalents.</li>
<li>Add links to the documentation to make nicer to explore the package from inside R</li>
<li>Implement a way to find markers for clusters exclusively upregulated</li>
<li>Refctor the code to make the coding nomenclature same as seurat</li>
</ol>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>Apache 2.0</li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>J. Sebastian Paez <br><small class="roles"> Author, maintainer </small>  </li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://travis-ci.org/jspaezp/sctree"><img src="https://travis-ci.org/jspaezp/sctree.svg?branch=master" alt="Travis build status"></a></li>
<li><a href="https://codecov.io/github/jspaezp/sctree?branch=master"><img src="https://codecov.io/gh/jspaezp/sctree/branch/master/graph/badge.svg" alt="Coverage status"></a></li>
<li><a href="https://joss.theoj.org/papers/4316d75e1e458003f7acfee08fd3922b"><img src="https://joss.theoj.org/papers/4316d75e1e458003f7acfee08fd3922b/status.svg" alt="status"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by J. Sebastian Paez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
