---
title: "sctree: a package to connect single cell rna-seq to biology using trees"
output: 
  github_document:
    toc: true
    toc_depth: 2
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

[![Travis build status](https://travis-ci.org/jspaezp/sctree.svg?branch=master)](https://travis-ci.org/jspaezp/sctree)
[![Coverage status](https://codecov.io/gh/jspaezp/sctree/branch/master/graph/badge.svg)](https://codecov.io/github/jspaezp/sctree?branch=master)

# sctree

The goal of sctree is to create a tool to accelerate the transition from
single cell rna-sequencing to calidation and new sub-population discovery.

Features suggesting pseudo-gating strategies to purify found populations via
flow-cytometry, antibody querying and cross validations between datasets.

Number of lines in roxygen comments:
`r system2("bash", "-c \"grep -RP '^#' ./R | wc -l\"", stdout = TRUE)`

Number of lines in R code:
`r system2("bash", "-c \"grep -RP '^[^#]' ./R | grep -vP '^$' | wc -l \"", stdout = TRUE)`


```{r child="vignettes/rmdchunks/_installation.Rmd"}
```

```{r child="vignettes/rmdchunks/_usage_intro.Rmd"}
```

```{r child="vignettes/rmdchunks/_finding_markers.Rmd"}
```

```{r child="vignettes/rmdchunks/_visualizing_markers.Rmd"}
```

```{r child="vignettes/rmdchunks/_suggesting_strategies.Rmd"}
```

```{r child="vignettes/rmdchunks/_antibodies.Rmd"}
```

```{r}
sessionInfo()
```


# Steps down the road

3. Address some of the TODO's in this repository
4. Reduce dependecies by replacing functions to base equivalents.
5. Add links to the documentation to make nicer to explore the package from inside R
6. Implement plot that actually illustrates the progressive gating in the decision tree
8. Implement a way to find markers for clusters exclusively upregulated
9. Refctor the code to make the coding nomenclature same as seurat


