<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>sctree: a package to connect single cell rna-seq to biology using trees • sctree</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script>




<meta property="og:title" content="sctree: a package to connect single cell rna-seq to biology using trees" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-title-body">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">sctree</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.4.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/sctree.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="articles/antibody_query.html">Query Antibody vendors from inside R</a>
    </li>
    <li>
      <a href="articles/plot_flowstyle.html">Flow style plotting</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>sctree: a package to connect single cell rna-seq to biology using trees</h1>
    </div>


<div id="sctree-a-package-to-connect-single-cell-rna-seq-to-biology-using-trees" class="section level1">

<!-- README.md is generated from README.Rmd. Please edit that file -->
</div>
<div id="sctree" class="section level1">
<h1 class="hasAnchor">
<a href="#sctree" class="anchor"></a>sctree</h1>
<p>The goal of sctree is to create a tool to accelerate the transition from single cell rna-sequencing to calidation and new sub-population discovery.</p>
<p>Features suggesting pseudo-gating strategies to purify found populations via flow-cytometry, antibody querying and cross validations between datasets.</p>
<p>Number of lines in roxygen comments: 0</p>
<p>Number of lines in R code: 0</p>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">git clone https<span class="op">:</span><span class="er">//</span>github.rcac.purdue.edu<span class="op">/</span>jpaezpae<span class="op">/</span>sctree sctree</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">cd sctree</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"></a>
<a class="sourceLine" id="cb1-4" data-line-number="4">R <span class="op">-</span>e <span class="st">"devtools::install('.')"</span></a></code></pre></div>
</div>
<div id="usage" class="section level1">
<h1 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h1>
<p>I am assuming you have already done your clustering and dimensional reduction using seurat and we have our seurat object.</p>
<p>For this examples we will use a dummy dataset that come bundled with the package</p>
<p><code>seurat_mix5050</code>, this dataset comes originally from the 1:1 mixture of Jurkat and 293T cells provided by 10x.</p>
<p>Original data can be found here:</p>
<ol>
<li><a href="https://support.10xgenomics.com/single-cell-gene-expression/datasets/1.1.0/jurkat:293t_50:50">1:1 mixture</a></li>
<li><a href="https://support.10xgenomics.com/single-cell-gene-expression/datasets/1.1.0/jurkat_293t_99_1">99:1 mixture</a></li>
</ol>
<!-- end list --><div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span>(sctree)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="co">#&gt; Loading required package: sctree</span></a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span>(Seurat)</a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="co">#&gt; Loading required package: Seurat</span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="co">#&gt; Loading required package: ggplot2</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="co">#&gt; Loading required package: cowplot</span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="co">#&gt; Attaching package: 'cowplot'</span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="co">#&gt; The following object is masked from 'package:ggplot2':</span></a>
<a class="sourceLine" id="cb2-10" data-line-number="10"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-11" data-line-number="11"><span class="co">#&gt;     ggsave</span></a>
<a class="sourceLine" id="cb2-12" data-line-number="12"><span class="co">#&gt; Loading required package: Matrix</span></a>
<a class="sourceLine" id="cb2-13" data-line-number="13"></a>
<a class="sourceLine" id="cb2-14" data-line-number="14"><span class="kw"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="dv">6</span>)</a>
<a class="sourceLine" id="cb2-15" data-line-number="15"></a>
<a class="sourceLine" id="cb2-16" data-line-number="16">seurat_mix5050 &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span>(<span class="st">"seurat_mix5050.RDS"</span>)</a>
<a class="sourceLine" id="cb2-17" data-line-number="17">seurat_mix9901 &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span>(<span class="st">"seurat_mix9901.RDS"</span>)</a>
<a class="sourceLine" id="cb2-18" data-line-number="18"></a>
<a class="sourceLine" id="cb2-19" data-line-number="19"><span class="kw"><a href="https://rdrr.io/pkg/Seurat/man/DimPlot.html">TSNEPlot</a></span>(seurat_mix9901)</a></code></pre></div>
<p><img src="sctree_readme_full_files/figure-gfm/unnamed-chunk-1-1.png"><!-- --></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/pkg/Seurat/man/DimPlot.html">TSNEPlot</a></span>(seurat_mix5050)</a></code></pre></div>
<p><img src="sctree_readme_full_files/figure-gfm/unnamed-chunk-1-2.png"><!-- --></p>
<div id="finding-important-variables-to-classify-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#finding-important-variables-to-classify-clusters" class="anchor"></a>Finding important variables to classify clusters</h2>
<p>We base our importances on the “classification value” they give to a random forest (using the implementation in the <code>ranger</code> package)</p>
<p>So lets fit the random forest … Here we are adding the <code>warn.imp.method</code> to prevent a warning message sent by <code>ranger</code> when most of the variables are correlated with the clustering.</p>
<p>Please reffer to the of the <code>importance_pvalues</code> section in the <a href="https://cran.r-project.org/web/packages/ranger/ranger.pdf">ranger documentation</a> when addressing this issue and for more details.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">rang_importances &lt;-<span class="st"> </span><span class="kw">ranger_importances.seurat</span>(</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">    seurat_mix5050,</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">    <span class="dt">cluster =</span> <span class="st">"ALL"</span>,</a>
<a class="sourceLine" id="cb4-4" data-line-number="4">    <span class="dt">warn.imp.method =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb4-5" data-line-number="5"></a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(rang_importances)</a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co">#&gt; [1] "ranger_fit"                "importances_ranger"       </span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8"><span class="co">#&gt; [3] "signif_importances_ranger"</span></a></code></pre></div>
<p>This gives us a list with 3 elements.</p>
<ol>
<li>The ranger fit object itself (handy if you want to inspect its classification accuracy)</li>
<li>The importance matrix deriven from ranger</li>
<li>A data frame containing only importances with pvalues under 0.05 (because biologists love p-values under 0.05)</li>
</ol>
<!-- end list --><div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">rang_importances[[<span class="dv">1</span>]]</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="co">#&gt; Ranger result</span></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="co">#&gt; Call:</span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="co">#&gt;  ranger::ranger(dependent.variable.name = "ident", data = data,      num.trees = num.trees, mtry = floor(ncol(data)/5), importance = importance,      classification = TRUE, ...) </span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="co">#&gt; Type:                             Classification </span></a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="co">#&gt; Number of trees:                  500 </span></a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="co">#&gt; Sample size:                      3388 </span></a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="co">#&gt; Number of independent variables:  538 </span></a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="co">#&gt; Mtry:                             107 </span></a>
<a class="sourceLine" id="cb5-12" data-line-number="12"><span class="co">#&gt; Target node size:                 1 </span></a>
<a class="sourceLine" id="cb5-13" data-line-number="13"><span class="co">#&gt; Variable importance mode:         impurity_corrected </span></a>
<a class="sourceLine" id="cb5-14" data-line-number="14"><span class="co">#&gt; Splitrule:                        gini </span></a>
<a class="sourceLine" id="cb5-15" data-line-number="15"><span class="co">#&gt; OOB prediction error:             0.24 %</span></a></code></pre></div>
<p>We can see that our classifier is 0.24 % accurate, as measured by its <em>OOB prediction error</em>.</p>
<p>Your classifier USUALLY will be just as good as your initial clustering is, therefore if your clustering is not all that consistent to start with, it is unrealistic that predicions will be done accurately.</p>
<p>Now … taking a look at the importance matrix we can see that a <code>pvalue</code> and a relative importance has been given to each gene.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(rang_importances[[<span class="dv">2</span>]])</a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="co">#&gt;                importance      pvalue</span></a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="co">#&gt; HES4           1.40955457 0.000000000</span></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="co">#&gt; ISG15          3.92726194 0.000000000</span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="co">#&gt; TNFRSF4        0.11080444 0.005780347</span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="co">#&gt; RP3-395M20.12 -0.02833254 0.965317919</span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="co">#&gt; KCNAB2         0.02682665 0.052023121</span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8"><span class="co">#&gt; CTNNBIP1      -0.03562707 0.988439306</span></a></code></pre></div>
<p>For simplicity we return by default a <code>data.frame</code> containing the ones with <code>pvalue &lt; 0.05</code></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(rang_importances[[<span class="dv">3</span>]])</a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="co">#&gt;         importance pvalue    gene</span></a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="co">#&gt; TMSB4X    141.5518      0  TMSB4X</span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="co">#&gt; ARHGDIB   139.4114      0 ARHGDIB</span></a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="co">#&gt; CKB       133.3990      0     CKB</span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="co">#&gt; XIST      132.2081      0    XIST</span></a>
<a class="sourceLine" id="cb7-7" data-line-number="7"><span class="co">#&gt; CDKN2A    119.2037      0  CDKN2A</span></a>
<a class="sourceLine" id="cb7-8" data-line-number="8"><span class="co">#&gt; ADA       111.5039      0     ADA</span></a></code></pre></div>
<p>Therefore, in this case we can say that the expression of the following genes would be usefull to form the clusters.</p>
</div>
<div id="visualizing-the-expected-outcome-of-a-flow-cytometry-experiment" class="section level2">
<h2 class="hasAnchor">
<a href="#visualizing-the-expected-outcome-of-a-flow-cytometry-experiment" class="anchor"></a>Visualizing the expected outcome of a flow cytometry experiment</h2>
<p>Lets say we choose the top 5 markers from the former list and we did a flow experiment … HYPOTHETICALLY the marker distribution would resemble the rna expression profile for which we have the function <code>plot_flowstyle</code></p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">top_markers &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(rang_importances[[<span class="dv">3</span>]]<span class="op">$</span>gene)</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">top_markers</a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="co">#&gt; [1] "TMSB4X"  "ARHGDIB" "CKB"     "XIST"    "CDKN2A"  "ADA"</span></a>
<a class="sourceLine" id="cb8-4" data-line-number="4">g &lt;-<span class="st"> </span><span class="kw"><a href="reference/plot_flowstyle.html">plot_flowstyle</a></span>(seurat_mix5050, <span class="dt">markernames =</span> top_markers)</a>
<a class="sourceLine" id="cb8-5" data-line-number="5">g</a></code></pre></div>
<p><img src="sctree_readme_full_files/figure-gfm/unnamed-chunk-6-1.png"><!-- --></p>
<p>Based on this, we can see that the red cluster in this plot is predominantly CD3+ ADA+ and ARHGDIB+, as well as ASNS-</p>
<p>We can also focus in one of the pannels (and check the color conventions)</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">g[<span class="dv">1</span>,<span class="dv">2</span>]</a></code></pre></div>
<p><img src="sctree_readme_full_files/figure-gfm/unnamed-chunk-7-1.png"><!-- --></p>
</div>
<div id="suggesting-a-gating-strategy-for-the-markers" class="section level2">
<h2 class="hasAnchor">
<a href="#suggesting-a-gating-strategy-for-the-markers" class="anchor"></a>Suggesting a gating strategy for the markers</h2>
<p>A general strategy to get separate all clusters</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"></a>
<a class="sourceLine" id="cb10-2" data-line-number="2">top_markers &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(rang_importances[[<span class="dv">3</span>]]<span class="op">$</span>gene)</a>
<a class="sourceLine" id="cb10-3" data-line-number="3"></a>
<a class="sourceLine" id="cb10-4" data-line-number="4">tree_fit &lt;-<span class="st"> </span><span class="kw"><a href="reference/fit_ctree.html">fit_ctree</a></span>(seurat_mix5050, <span class="dt">genes_use =</span> top_markers, <span class="dt">cluster =</span> <span class="st">"ALL"</span>)</a></code></pre></div>
<p>Visualizing the tree as … a tree … we can see how our model is a simple series of yes/no questions.</p>
<p>If we wanted to classifiy a random cell: in the first <code>node</code>, we check if the expression of that gene is higher or lower than a given value, if it is lower, we proceed to the left, if not we go right. We keep doing that until we have no more <code>branches</code>. This final node will have a predicted cluster, in this plot we can also see how pure can we expect this group to be and how many of the cells in our training set clasify as part of it.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(tree_fit)</a></code></pre></div>
<p><img src="sctree_readme_full_files/figure-gfm/unnamed-chunk-9-1.png"><!-- --></p>
<p>When inspecting the tree_fit, we can see a more detailed text representation of this tree.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/pkg/Seurat/man/print.DimReduc.html">print</a></span>(tree_fit)</a>
<a class="sourceLine" id="cb12-2" data-line-number="2"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb12-3" data-line-number="3"><span class="co">#&gt; Model formula:</span></a>
<a class="sourceLine" id="cb12-4" data-line-number="4"><span class="co">#&gt; ident ~ TMSB4X + ARHGDIB + CKB + XIST + CDKN2A + ADA</span></a>
<a class="sourceLine" id="cb12-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb12-6" data-line-number="6"><span class="co">#&gt; Fitted party:</span></a>
<a class="sourceLine" id="cb12-7" data-line-number="7"><span class="co">#&gt; [1] root</span></a>
<a class="sourceLine" id="cb12-8" data-line-number="8"><span class="co">#&gt; |   [2] CKB &lt;= 1.32554</span></a>
<a class="sourceLine" id="cb12-9" data-line-number="9"><span class="co">#&gt; |   |   [3] XIST &lt;= 1.12625</span></a>
<a class="sourceLine" id="cb12-10" data-line-number="10"><span class="co">#&gt; |   |   |   [4] ARHGDIB &lt;= 1.20526: 0 (n = 18, err = 5.6%)</span></a>
<a class="sourceLine" id="cb12-11" data-line-number="11"><span class="co">#&gt; |   |   |   [5] ARHGDIB &gt; 1.20526: 0 (n = 1734, err = 0.0%)</span></a>
<a class="sourceLine" id="cb12-12" data-line-number="12"><span class="co">#&gt; |   |   [6] XIST &gt; 1.12625: 1 (n = 18, err = 50.0%)</span></a>
<a class="sourceLine" id="cb12-13" data-line-number="13"><span class="co">#&gt; |   [7] CKB &gt; 1.32554</span></a>
<a class="sourceLine" id="cb12-14" data-line-number="14"><span class="co">#&gt; |   |   [8] ARHGDIB &lt;= 1.29467</span></a>
<a class="sourceLine" id="cb12-15" data-line-number="15"><span class="co">#&gt; |   |   |   [9] XIST &lt;= 0</span></a>
<a class="sourceLine" id="cb12-16" data-line-number="16"><span class="co">#&gt; |   |   |   |   [10] TMSB4X &lt;= 0.7876: 3 (n = 19, err = 31.6%)</span></a>
<a class="sourceLine" id="cb12-17" data-line-number="17"><span class="co">#&gt; |   |   |   |   [11] TMSB4X &gt; 0.7876: 3 (n = 7, err = 28.6%)</span></a>
<a class="sourceLine" id="cb12-18" data-line-number="18"><span class="co">#&gt; |   |   |   [12] XIST &gt; 0</span></a>
<a class="sourceLine" id="cb12-19" data-line-number="19"><span class="co">#&gt; |   |   |   |   [13] TMSB4X &lt;= 1.77781</span></a>
<a class="sourceLine" id="cb12-20" data-line-number="20"><span class="co">#&gt; |   |   |   |   |   [14] ARHGDIB &lt;= 0.87204</span></a>
<a class="sourceLine" id="cb12-21" data-line-number="21"><span class="co">#&gt; |   |   |   |   |   |   [15] XIST &lt;= 1.23346</span></a>
<a class="sourceLine" id="cb12-22" data-line-number="22"><span class="co">#&gt; |   |   |   |   |   |   |   [16] CDKN2A &lt;= 1.92909: 1 (n = 25, err = 4.0%)</span></a>
<a class="sourceLine" id="cb12-23" data-line-number="23"><span class="co">#&gt; |   |   |   |   |   |   |   [17] CDKN2A &gt; 1.92909: 1 (n = 7, err = 28.6%)</span></a>
<a class="sourceLine" id="cb12-24" data-line-number="24"><span class="co">#&gt; |   |   |   |   |   |   [18] XIST &gt; 1.23346</span></a>
<a class="sourceLine" id="cb12-25" data-line-number="25"><span class="co">#&gt; |   |   |   |   |   |   |   [19] ARHGDIB &lt;= 0.69905: 1 (n = 1437, err = 0.0%)</span></a>
<a class="sourceLine" id="cb12-26" data-line-number="26"><span class="co">#&gt; |   |   |   |   |   |   |   [20] ARHGDIB &gt; 0.69905: 1 (n = 30, err = 3.3%)</span></a>
<a class="sourceLine" id="cb12-27" data-line-number="27"><span class="co">#&gt; |   |   |   |   |   [21] ARHGDIB &gt; 0.87204</span></a>
<a class="sourceLine" id="cb12-28" data-line-number="28"><span class="co">#&gt; |   |   |   |   |   |   [22] TMSB4X &lt;= 0.8868: 1 (n = 19, err = 0.0%)</span></a>
<a class="sourceLine" id="cb12-29" data-line-number="29"><span class="co">#&gt; |   |   |   |   |   |   [23] TMSB4X &gt; 0.8868: 1 (n = 7, err = 42.9%)</span></a>
<a class="sourceLine" id="cb12-30" data-line-number="30"><span class="co">#&gt; |   |   |   |   [24] TMSB4X &gt; 1.77781: 2 (n = 16, err = 50.0%)</span></a>
<a class="sourceLine" id="cb12-31" data-line-number="31"><span class="co">#&gt; |   |   [25] ARHGDIB &gt; 1.29467</span></a>
<a class="sourceLine" id="cb12-32" data-line-number="32"><span class="co">#&gt; |   |   |   [26] CDKN2A &lt;= 0.73944: 0 (n = 7, err = 28.6%)</span></a>
<a class="sourceLine" id="cb12-33" data-line-number="33"><span class="co">#&gt; |   |   |   [27] CDKN2A &gt; 0.73944: 2 (n = 44, err = 0.0%)</span></a>
<a class="sourceLine" id="cb12-34" data-line-number="34"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb12-35" data-line-number="35"><span class="co">#&gt; Number of inner nodes:    13</span></a>
<a class="sourceLine" id="cb12-36" data-line-number="36"><span class="co">#&gt; Number of terminal nodes: 14</span></a></code></pre></div>
<p>Sometimes one might think that the proposed strategy is too complicated or not implementable in the experimental settings, in order to add constrians to the fit one can give additional arguments that will be passed to <code><a href="https://rdrr.io/pkg/partykit/man/ctree_control.html">partykit::ctree_control</a></code>, such as <code>maxdepth = 2</code> (maximum 2 questions per cell)</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">tree_fit &lt;-<span class="st"> </span><span class="kw"><a href="reference/fit_ctree.html">fit_ctree</a></span>(</a>
<a class="sourceLine" id="cb13-2" data-line-number="2">  seurat_mix5050, <span class="dt">genes_use =</span> top_markers, </a>
<a class="sourceLine" id="cb13-3" data-line-number="3">  <span class="dt">cluster =</span> <span class="st">"ALL"</span>, <span class="dt">maxdepth =</span> <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb13-4" data-line-number="4"><span class="kw"><a href="https://rdrr.io/pkg/Seurat/man/print.DimReduc.html">print</a></span>(tree_fit)</a>
<a class="sourceLine" id="cb13-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb13-6" data-line-number="6"><span class="co">#&gt; Model formula:</span></a>
<a class="sourceLine" id="cb13-7" data-line-number="7"><span class="co">#&gt; ident ~ TMSB4X + ARHGDIB + CKB + XIST + CDKN2A + ADA</span></a>
<a class="sourceLine" id="cb13-8" data-line-number="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb13-9" data-line-number="9"><span class="co">#&gt; Fitted party:</span></a>
<a class="sourceLine" id="cb13-10" data-line-number="10"><span class="co">#&gt; [1] root</span></a>
<a class="sourceLine" id="cb13-11" data-line-number="11"><span class="co">#&gt; |   [2] CKB &lt;= 1.32554</span></a>
<a class="sourceLine" id="cb13-12" data-line-number="12"><span class="co">#&gt; |   |   [3] XIST &lt;= 1.12625: 0 (n = 1752, err = 0.1%)</span></a>
<a class="sourceLine" id="cb13-13" data-line-number="13"><span class="co">#&gt; |   |   [4] XIST &gt; 1.12625: 1 (n = 18, err = 50.0%)</span></a>
<a class="sourceLine" id="cb13-14" data-line-number="14"><span class="co">#&gt; |   [5] CKB &gt; 1.32554</span></a>
<a class="sourceLine" id="cb13-15" data-line-number="15"><span class="co">#&gt; |   |   [6] ARHGDIB &lt;= 1.29467: 1 (n = 1567, err = 2.2%)</span></a>
<a class="sourceLine" id="cb13-16" data-line-number="16"><span class="co">#&gt; |   |   [7] ARHGDIB &gt; 1.29467: 2 (n = 51, err = 9.8%)</span></a>
<a class="sourceLine" id="cb13-17" data-line-number="17"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb13-18" data-line-number="18"><span class="co">#&gt; Number of inner nodes:    3</span></a>
<a class="sourceLine" id="cb13-19" data-line-number="19"><span class="co">#&gt; Number of terminal nodes: 4</span></a>
<a class="sourceLine" id="cb13-20" data-line-number="20"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(tree_fit)</a></code></pre></div>
<p><img src="sctree_readme_full_files/figure-gfm/unnamed-chunk-11-1.png"><!-- --></p>
<p>Since not all variables are ultimately used in our classifier, one can acces the ones that were by using <code>varimp(tree_fit)</code></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">partykit<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/partykit/man/varimp.html">varimp</a></span>(tree_fit)</a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="co">#&gt;        CKB       XIST    ARHGDIB </span></a>
<a class="sourceLine" id="cb14-3" data-line-number="3"><span class="co">#&gt; 0.82866521 0.03625885 0.33081234</span></a>
<a class="sourceLine" id="cb14-4" data-line-number="4"><span class="kw"><a href="reference/plot_flowstyle.html">plot_flowstyle</a></span>(seurat_mix5050, <span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(partykit<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/partykit/man/varimp.html">varimp</a></span>(tree_fit)))</a></code></pre></div>
<p><img src="sctree_readme_full_files/figure-gfm/unnamed-chunk-12-1.png"><!-- --></p>
<p>One can also request the package to suggest a specific strategy only for a given cluster. This function is not expected to give drastically different results in datasets with few clusters, but it can definitely come usefull when many clusters are present and one is interested in a specific one.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">tree_fit &lt;-<span class="st"> </span><span class="kw"><a href="reference/fit_ctree.html">fit_ctree</a></span>(seurat_mix5050, <span class="dt">genes_use =</span> top_markers, <span class="dt">cluster =</span> <span class="st">"0"</span>)</a>
<a class="sourceLine" id="cb15-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/pkg/Seurat/man/print.DimReduc.html">print</a></span>(tree_fit)</a>
<a class="sourceLine" id="cb15-3" data-line-number="3"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb15-4" data-line-number="4"><span class="co">#&gt; Model formula:</span></a>
<a class="sourceLine" id="cb15-5" data-line-number="5"><span class="co">#&gt; ident ~ TMSB4X + ARHGDIB + CKB + XIST + CDKN2A + ADA</span></a>
<a class="sourceLine" id="cb15-6" data-line-number="6"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb15-7" data-line-number="7"><span class="co">#&gt; Fitted party:</span></a>
<a class="sourceLine" id="cb15-8" data-line-number="8"><span class="co">#&gt; [1] root</span></a>
<a class="sourceLine" id="cb15-9" data-line-number="9"><span class="co">#&gt; |   [2] CKB &lt;= 1.32554</span></a>
<a class="sourceLine" id="cb15-10" data-line-number="10"><span class="co">#&gt; |   |   [3] XIST &lt;= 1.12625</span></a>
<a class="sourceLine" id="cb15-11" data-line-number="11"><span class="co">#&gt; |   |   |   [4] ARHGDIB &lt;= 1.20526: indeed clus 0 (n = 18, err = 5.6%)</span></a>
<a class="sourceLine" id="cb15-12" data-line-number="12"><span class="co">#&gt; |   |   |   [5] ARHGDIB &gt; 1.20526: indeed clus 0 (n = 1734, err = 0.0%)</span></a>
<a class="sourceLine" id="cb15-13" data-line-number="13"><span class="co">#&gt; |   |   [6] XIST &gt; 1.12625: not clus 0 (n = 18, err = 33.3%)</span></a>
<a class="sourceLine" id="cb15-14" data-line-number="14"><span class="co">#&gt; |   [7] CKB &gt; 1.32554</span></a>
<a class="sourceLine" id="cb15-15" data-line-number="15"><span class="co">#&gt; |   |   [8] ARHGDIB &lt;= 1.9863</span></a>
<a class="sourceLine" id="cb15-16" data-line-number="16"><span class="co">#&gt; |   |   |   [9] TMSB4X &lt;= 2.72634: not clus 0 (n = 1584, err = 0.0%)</span></a>
<a class="sourceLine" id="cb15-17" data-line-number="17"><span class="co">#&gt; |   |   |   [10] TMSB4X &gt; 2.72634: not clus 0 (n = 17, err = 5.9%)</span></a>
<a class="sourceLine" id="cb15-18" data-line-number="18"><span class="co">#&gt; |   |   [11] ARHGDIB &gt; 1.9863: not clus 0 (n = 17, err = 29.4%)</span></a>
<a class="sourceLine" id="cb15-19" data-line-number="19"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb15-20" data-line-number="20"><span class="co">#&gt; Number of inner nodes:    5</span></a>
<a class="sourceLine" id="cb15-21" data-line-number="21"><span class="co">#&gt; Number of terminal nodes: 6</span></a></code></pre></div>
</div>
<div id="finding-equivalent-clusters-in-two-datasets" class="section level2">
<h2 class="hasAnchor">
<a href="#finding-equivalent-clusters-in-two-datasets" class="anchor"></a>Finding equivalent clusters in two datasets</h2>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(seurat_mix9901)</a>
<a class="sourceLine" id="cb16-2" data-line-number="2"><span class="co">#&gt; Warning in data(seurat_mix9901): data set 'seurat_mix9901' not found</span></a>
<a class="sourceLine" id="cb16-3" data-line-number="3">seurat_mix9901</a>
<a class="sourceLine" id="cb16-4" data-line-number="4"><span class="co">#&gt; An object of class seurat in project SeuratProject </span></a>
<a class="sourceLine" id="cb16-5" data-line-number="5"><span class="co">#&gt;  32738 genes across 4185 samples.</span></a></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">validation_results &lt;-<span class="st"> </span><span class="kw"><a href="reference/cross_validate.html">cross_validate</a></span>(</a>
<a class="sourceLine" id="cb17-2" data-line-number="2">    seurat_mix5050, seurat_mix9901, </a>
<a class="sourceLine" id="cb17-3" data-line-number="3">    <span class="dt">cluster =</span> <span class="st">"ALL"</span>)</a>
<a class="sourceLine" id="cb17-4" data-line-number="4"><span class="co">#&gt; Warning in ranger::importance_pvalues(ranger_fit): Only few negative</span></a>
<a class="sourceLine" id="cb17-5" data-line-number="5"><span class="co">#&gt; importance values found, inaccurate p-values. Consider the 'altmann'</span></a>
<a class="sourceLine" id="cb17-6" data-line-number="6"><span class="co">#&gt; approach.</span></a>
<a class="sourceLine" id="cb17-7" data-line-number="7"><span class="co">#&gt; Warning in max(elems_per_rule): no non-missing arguments to max; returning</span></a>
<a class="sourceLine" id="cb17-8" data-line-number="8"><span class="co">#&gt; -Inf</span></a>
<a class="sourceLine" id="cb17-9" data-line-number="9"></a>
<a class="sourceLine" id="cb17-10" data-line-number="10"><span class="co">#&gt; Warning in max(elems_per_rule): no non-missing arguments to max; returning</span></a>
<a class="sourceLine" id="cb17-11" data-line-number="11"><span class="co">#&gt; -Inf</span></a>
<a class="sourceLine" id="cb17-12" data-line-number="12"></a>
<a class="sourceLine" id="cb17-13" data-line-number="13"><span class="co">#&gt; Warning in max(elems_per_rule): no non-missing arguments to max; returning</span></a>
<a class="sourceLine" id="cb17-14" data-line-number="14"><span class="co">#&gt; -Inf</span></a>
<a class="sourceLine" id="cb17-15" data-line-number="15"></a>
<a class="sourceLine" id="cb17-16" data-line-number="16">validation_results[[<span class="dv">1</span>]]</a>
<a class="sourceLine" id="cb17-17" data-line-number="17"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb17-18" data-line-number="18"><span class="co">#&gt; Model formula:</span></a>
<a class="sourceLine" id="cb17-19" data-line-number="19"><span class="co">#&gt; ident ~ CKB + TMSB4X + CD3D + EIF5A + XIST + ARHGDIB + CDKN2A + </span></a>
<a class="sourceLine" id="cb17-20" data-line-number="20"><span class="co">#&gt;     ADA + MZB1 + OCIAD2 + LCK + NUCB2 + RPS4X + GAPDH + ITM2A + </span></a>
<a class="sourceLine" id="cb17-21" data-line-number="21"><span class="co">#&gt;     CA2 + RPL29 + CD3G + BEX1 + PSMB8 + GMFG + FTH1 + LEF1 + </span></a>
<a class="sourceLine" id="cb17-22" data-line-number="22"><span class="co">#&gt;     CBR1 + RNASEH2B + GAL + GTF3A + TSPO + `MT-CO3` + `MT-CO1` + </span></a>
<a class="sourceLine" id="cb17-23" data-line-number="23"><span class="co">#&gt;     CD1E + NT5C3B + CHCHD10 + FYB + ISG15 + RAB13 + DDT + CALD1 + </span></a>
<a class="sourceLine" id="cb17-24" data-line-number="24"><span class="co">#&gt;     RPL12 + DMKN + MDK + RPL26 + TUBB + MALAT1 + TYMS + SH3BGRL3 + </span></a>
<a class="sourceLine" id="cb17-25" data-line-number="25"><span class="co">#&gt;     CDK6 + MAP1A + PTPN7 + PCSK1N + HEY1 + CHCHD5 + AIF1 + UCHL1 + </span></a>
<a class="sourceLine" id="cb17-26" data-line-number="26"><span class="co">#&gt;     EIF4EBP1 + C3orf14 + PRDX4 + CALR + RAB34 + HES4 + GYPC + </span></a>
<a class="sourceLine" id="cb17-27" data-line-number="27"><span class="co">#&gt;     PCNA + SELM + HOXA10 + IGLL1 + RBBP7 + UBE2L6 + CDC42EP1 + </span></a>
<a class="sourceLine" id="cb17-28" data-line-number="28"><span class="co">#&gt;     PGK1 + HSP90AA1 + RPL36A + CHI3L2 + FKBP10 + HOXA9 + TUBA1B + </span></a>
<a class="sourceLine" id="cb17-29" data-line-number="29"><span class="co">#&gt;     ZNF703 + GLO1 + DNAJA1 + UBC + RAC3 + `HHIP-AS1` + ARPP21 + </span></a>
<a class="sourceLine" id="cb17-30" data-line-number="30"><span class="co">#&gt;     NME3 + CD63 + B2M + TSTD1 + CD3E + GLRX + TCF7 + NEURL1 + </span></a>
<a class="sourceLine" id="cb17-31" data-line-number="31"><span class="co">#&gt;     CORO1A + RHOH + ACTB + SPN + PLS3 + ID2 + RPP25 + S100A11 + </span></a>
<a class="sourceLine" id="cb17-32" data-line-number="32"><span class="co">#&gt;     STMN1 + TFDP2 + RAB40B + ECI1 + MORF4L2 + RPLP1 + CCNB2 + </span></a>
<a class="sourceLine" id="cb17-33" data-line-number="33"><span class="co">#&gt;     MGMT + CD151 + PGRMC1 + GSDMD + AK1 + S100A13 + TCEAL4 + </span></a>
<a class="sourceLine" id="cb17-34" data-line-number="34"><span class="co">#&gt;     PTTG1 + AC002454.1 + TFAP2A + KRT18 + TUBB4B + MYO1G + SLIT2 + </span></a>
<a class="sourceLine" id="cb17-35" data-line-number="35"><span class="co">#&gt;     STRA13 + HSP90B1 + MYB + FTL + ENAH + ARF1 + PODXL2 + CNN3 + </span></a>
<a class="sourceLine" id="cb17-36" data-line-number="36"><span class="co">#&gt;     ACTG1 + VAMP5 + SHQ1 + CDK1 + HSD17B12 + NAA10 + CD1B + DCAF13 + </span></a>
<a class="sourceLine" id="cb17-37" data-line-number="37"><span class="co">#&gt;     PRDX6 + HSPA1B + ZNF503 + SH2D1A + PERP + MRP63 + LCP1 + </span></a>
<a class="sourceLine" id="cb17-38" data-line-number="38"><span class="co">#&gt;     CXCR4 + SIVA1 + VIM + PPP1R18 + CKLF + ASF1A + ZNF22 + TPTEP1 + </span></a>
<a class="sourceLine" id="cb17-39" data-line-number="39"><span class="co">#&gt;     CENPF + MT2A + USP14 + ANXA2 + PSMB9 + PYGL + PSME1 + CAPN2 + </span></a>
<a class="sourceLine" id="cb17-40" data-line-number="40"><span class="co">#&gt;     SYNGR1 + PPIC + MCM5 + EPM2AIP1 + HSPA8 + CSNK1E + IFI16 + </span></a>
<a class="sourceLine" id="cb17-41" data-line-number="41"><span class="co">#&gt;     SCML1 + ME2 + ACAT2 + BTG1 + MAP1B + SMC2 + IL2RG + CST3 + </span></a>
<a class="sourceLine" id="cb17-42" data-line-number="42"><span class="co">#&gt;     FUCA2 + PFN2 + LINC00632 + FABP5 + LPXN + DEPDC1 + ACSF3 + </span></a>
<a class="sourceLine" id="cb17-43" data-line-number="43"><span class="co">#&gt;     CEP70 + NDFIP2 + BIRC5 + CARHSP1 + DNAJC15 + COA3 + NMRAL1 + </span></a>
<a class="sourceLine" id="cb17-44" data-line-number="44"><span class="co">#&gt;     DEGS1 + KPNA2 + AEBP1 + MLF1 + CBX5 + RAC2 + HOTAIRM1 + EIF5AL1 + </span></a>
<a class="sourceLine" id="cb17-45" data-line-number="45"><span class="co">#&gt;     HNRNPF + TNFRSF4 + TIMP1 + NMI + C14orf64 + RFC4 + LDLRAD4 + </span></a>
<a class="sourceLine" id="cb17-46" data-line-number="46"><span class="co">#&gt;     KIAA1430 + CBS + KNSTRN + MTAP + C12orf65 + HOXB6 + SEPW1 + </span></a>
<a class="sourceLine" id="cb17-47" data-line-number="47"><span class="co">#&gt;     CDC20 + DPP7 + CSRP2 + PPP2CA + KCNAB2 + NEAT1 + TPX2 + MFNG + </span></a>
<a class="sourceLine" id="cb17-48" data-line-number="48"><span class="co">#&gt;     PTPRCAP + RRM2 + ARHGAP30 + PINLYP + CKS2 + FAM127B + TM7SF3 + </span></a>
<a class="sourceLine" id="cb17-49" data-line-number="49"><span class="co">#&gt;     FBXO25 + HIST1H4C + SKP1 + NME4 + LAPTM5 + XBP1 + FHL1 + </span></a>
<a class="sourceLine" id="cb17-50" data-line-number="50"><span class="co">#&gt;     GNA15 + TRAF3IP3 + CCNB1 + TSC22D3 + TMSB10 + CD1D + ANXA5 + </span></a>
<a class="sourceLine" id="cb17-51" data-line-number="51"><span class="co">#&gt;     CCND3 + SRSF6 + MSN + SELL + HIST1H2BK + C12orf57 + FNBP1 + </span></a>
<a class="sourceLine" id="cb17-52" data-line-number="52"><span class="co">#&gt;     TCTEX1D2 + MYO7B + IDH2 + CENPE + LIN28B + NAPRT1 + ITGA4 + </span></a>
<a class="sourceLine" id="cb17-53" data-line-number="53"><span class="co">#&gt;     FDFT1 + NUF2 + EPDR1 + ANKRD12 + HCLS1 + MYC + SPTBN1 + KIAA0226L + </span></a>
<a class="sourceLine" id="cb17-54" data-line-number="54"><span class="co">#&gt;     EMP3 + HMGB3 + BID + TSC22D1 + FSCN1 + AES + JUN + KIAA0020 + </span></a>
<a class="sourceLine" id="cb17-55" data-line-number="55"><span class="co">#&gt;     GPSM3 + SLA + PLP2 + CDCA8 + ATP6V1H + OXCT1 + MT1X + TEX30 + </span></a>
<a class="sourceLine" id="cb17-56" data-line-number="56"><span class="co">#&gt;     TK1 + ICAM2 + S100A10 + PAK2 + SOX4 + ARID5B + NUDT8 + PCID2 + </span></a>
<a class="sourceLine" id="cb17-57" data-line-number="57"><span class="co">#&gt;     ALDH1A2 + BAG2 + ID4 + RASGRP2 + `RP11-973H7.4` + CITED1 + </span></a>
<a class="sourceLine" id="cb17-58" data-line-number="58"><span class="co">#&gt;     ZAP70 + HENMT1 + DNTT + `PRKAG2-AS1` + CD247 + DHRS4 + ATP6AP1L + </span></a>
<a class="sourceLine" id="cb17-59" data-line-number="59"><span class="co">#&gt;     SEPT1 + FES + ELOVL5 + NDN + MAL + SGOL2 + CDKN2AIP + NUSAP1 + </span></a>
<a class="sourceLine" id="cb17-60" data-line-number="60"><span class="co">#&gt;     AURKA + SLC43A3 + RAP1GDS1 + CELF2 + SELPLG + PSME2 + CFL2 + </span></a>
<a class="sourceLine" id="cb17-61" data-line-number="61"><span class="co">#&gt;     CD7 + IER2 + EIF4A3 + PBK + JAKMIP1 + GRAP2 + VAMP8 + CTSG + </span></a>
<a class="sourceLine" id="cb17-62" data-line-number="62"><span class="co">#&gt;     BEX2 + HIST2H2AC + SMC1B + TBC1D10C + RCSD1 + CTNNBIP1 + </span></a>
<a class="sourceLine" id="cb17-63" data-line-number="63"><span class="co">#&gt;     TUBA1A + EVL + SIT1 + FXYD5 + HMGB2 + RHOB + FERMT3 + CPNE1 + </span></a>
<a class="sourceLine" id="cb17-64" data-line-number="64"><span class="co">#&gt;     APLP2 + CENPA + CD47 + DUSP1 + C5orf56 + SPAG5 + ICAM3 + </span></a>
<a class="sourceLine" id="cb17-65" data-line-number="65"><span class="co">#&gt;     ETHE1 + CMPK2 + PTPRC + CASP8 + PSAT1 + CCZ1B + RAG1 + RBM3 + </span></a>
<a class="sourceLine" id="cb17-66" data-line-number="66"><span class="co">#&gt;     `RP11-1094M14.11`</span></a>
<a class="sourceLine" id="cb17-67" data-line-number="67"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb17-68" data-line-number="68"><span class="co">#&gt; Fitted party:</span></a>
<a class="sourceLine" id="cb17-69" data-line-number="69"><span class="co">#&gt; [1] root</span></a>
<a class="sourceLine" id="cb17-70" data-line-number="70"><span class="co">#&gt; |   [2] CD3D &lt;= 1.23381</span></a>
<a class="sourceLine" id="cb17-71" data-line-number="71"><span class="co">#&gt; |   |   [3] OCIAD2 &lt;= 0.90609: 1 (n = 1565, err = 1.8%)</span></a>
<a class="sourceLine" id="cb17-72" data-line-number="72"><span class="co">#&gt; |   |   [4] OCIAD2 &gt; 0.90609: 0 (n = 16, err = 18.8%)</span></a>
<a class="sourceLine" id="cb17-73" data-line-number="73"><span class="co">#&gt; |   [5] CD3D &gt; 1.23381</span></a>
<a class="sourceLine" id="cb17-74" data-line-number="74"><span class="co">#&gt; |   |   [6] XIST &lt;= 1.20355</span></a>
<a class="sourceLine" id="cb17-75" data-line-number="75"><span class="co">#&gt; |   |   |   [7] GAL &lt;= 0.76111: 0 (n = 1734, err = 0.1%)</span></a>
<a class="sourceLine" id="cb17-76" data-line-number="76"><span class="co">#&gt; |   |   |   [8] GAL &gt; 0.76111: 0 (n = 18, err = 33.3%)</span></a>
<a class="sourceLine" id="cb17-77" data-line-number="77"><span class="co">#&gt; |   |   [9] XIST &gt; 1.20355</span></a>
<a class="sourceLine" id="cb17-78" data-line-number="78"><span class="co">#&gt; |   |   |   [10] ARHGAP30 &lt;= 0.38927</span></a>
<a class="sourceLine" id="cb17-79" data-line-number="79"><span class="co">#&gt; |   |   |   |   [11] EIF5A &lt;= 2.09015: 2 (n = 11, err = 27.3%)</span></a>
<a class="sourceLine" id="cb17-80" data-line-number="80"><span class="co">#&gt; |   |   |   |   [12] EIF5A &gt; 2.09015: 2 (n = 37, err = 0.0%)</span></a>
<a class="sourceLine" id="cb17-81" data-line-number="81"><span class="co">#&gt; |   |   |   [13] ARHGAP30 &gt; 0.38927: 0 (n = 7, err = 42.9%)</span></a>
<a class="sourceLine" id="cb17-82" data-line-number="82"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb17-83" data-line-number="83"><span class="co">#&gt; Number of inner nodes:    6</span></a>
<a class="sourceLine" id="cb17-84" data-line-number="84"><span class="co">#&gt; Number of terminal nodes: 7</span></a></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1">validation_results<span class="op">$</span>confusion_matrix</a>
<a class="sourceLine" id="cb18-2" data-line-number="2"><span class="co">#&gt;          cluster</span></a>
<a class="sourceLine" id="cb18-3" data-line-number="3"><span class="co">#&gt; predicted    0    1    2    3    4</span></a>
<a class="sourceLine" id="cb18-4" data-line-number="4"><span class="co">#&gt;         0 1814 1068  945  327    1</span></a>
<a class="sourceLine" id="cb18-5" data-line-number="5"><span class="co">#&gt;         1    0    0    1    0   27</span></a>
<a class="sourceLine" id="cb18-6" data-line-number="6"><span class="co">#&gt;         2    0    2    0    0    0</span></a>
<a class="sourceLine" id="cb18-7" data-line-number="7"><span class="co">#&gt;         3    0    0    0    0    0</span></a></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">freq_matrix &lt;-<span class="st"> </span><span class="kw"><a href="reference/as.frequency.matrix.html">as.frequency.matrix</a></span>(validation_results<span class="op">$</span>confusion_matrix)</a>
<a class="sourceLine" id="cb19-2" data-line-number="2">freq_matrix</a>
<a class="sourceLine" id="cb19-3" data-line-number="3"><span class="co">#&gt;          cluster</span></a>
<a class="sourceLine" id="cb19-4" data-line-number="4"><span class="co">#&gt; predicted           0           1           2           3           4</span></a>
<a class="sourceLine" id="cb19-5" data-line-number="5"><span class="co">#&gt;         0 100.0000000  99.8130841  99.8942918 100.0000000   3.5714286</span></a>
<a class="sourceLine" id="cb19-6" data-line-number="6"><span class="co">#&gt;         1   0.0000000   0.0000000   0.1057082   0.0000000  96.4285714</span></a>
<a class="sourceLine" id="cb19-7" data-line-number="7"><span class="co">#&gt;         2   0.0000000   0.1869159   0.0000000   0.0000000   0.0000000</span></a>
<a class="sourceLine" id="cb19-8" data-line-number="8"><span class="co">#&gt;         3   0.0000000   0.0000000   0.0000000   0.0000000   0.0000000</span></a></code></pre></div>
<p>This would mean that cells in cluster 0 in the <code>seurat_mix9901</code> dataset are classified 99.23 % of the time as bleonging to cluster 0 of the <code>seurat_mix5050</code></p>
<p>These frequencies can be visualized by plotting them in a heatmap</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"></a>
<a class="sourceLine" id="cb20-2" data-line-number="2"><span class="kw"><a href="reference/autoplot.table.html">autoplot.table</a></span>(freq_matrix,</a>
<a class="sourceLine" id="cb20-3" data-line-number="3">               <span class="dt">min_color =</span> <span class="dv">50</span>,</a>
<a class="sourceLine" id="cb20-4" data-line-number="4">               <span class="dt">show_number =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb20-5" data-line-number="5"><span class="co">#&gt; Warning: Removed 15 rows containing missing values (geom_text).</span></a></code></pre></div>
<p><img src="sctree_readme_full_files/figure-gfm/unnamed-chunk-18-1.png"><!-- --></p>
<p>Here we can see that in the <em>9901</em> dataset (predicted), both clusters 0 and 1 are classified mostrly as cluster 0 in the <em>5050</em> dataset, while the cluster 2 is mainly classified as 1.</p>
<p>(remember that the numbers are arbitrary and only mean something within each dataset)</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1">validation_results[[<span class="dv">3</span>]]</a>
<a class="sourceLine" id="cb21-2" data-line-number="2"><span class="co">#&gt; $`0`</span></a>
<a class="sourceLine" id="cb21-3" data-line-number="3"><span class="co">#&gt; $`0`$all</span></a>
<a class="sourceLine" id="cb21-4" data-line-number="4"><span class="co">#&gt; [1] "CD3D &gt; 1.23380572030749"</span></a>
<a class="sourceLine" id="cb21-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb21-6" data-line-number="6"><span class="co">#&gt; $`0`$majority</span></a>
<a class="sourceLine" id="cb21-7" data-line-number="7"><span class="co">#&gt; [1] "XIST &lt;= 1.20354519639701" "GAL &lt;= 0.761105335702687"</span></a>
<a class="sourceLine" id="cb21-8" data-line-number="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb21-9" data-line-number="9"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb21-10" data-line-number="10"><span class="co">#&gt; $`1`</span></a>
<a class="sourceLine" id="cb21-11" data-line-number="11"><span class="co">#&gt; $`1`$all</span></a>
<a class="sourceLine" id="cb21-12" data-line-number="12"><span class="co">#&gt; [1] "CD3D &lt;= 1.23380572030749"    "OCIAD2 &lt;= 0.906093418104814"</span></a>
<a class="sourceLine" id="cb21-13" data-line-number="13"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb21-14" data-line-number="14"><span class="co">#&gt; $`1`$majority</span></a>
<a class="sourceLine" id="cb21-15" data-line-number="15"><span class="co">#&gt; character(0)</span></a>
<a class="sourceLine" id="cb21-16" data-line-number="16"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb21-17" data-line-number="17"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb21-18" data-line-number="18"><span class="co">#&gt; $`2`</span></a>
<a class="sourceLine" id="cb21-19" data-line-number="19"><span class="co">#&gt; $`2`$all</span></a>
<a class="sourceLine" id="cb21-20" data-line-number="20"><span class="co">#&gt; [1] "CD3D &gt; 1.23380572030749"       "XIST &gt; 1.20354519639701"      </span></a>
<a class="sourceLine" id="cb21-21" data-line-number="21"><span class="co">#&gt; [3] "ARHGAP30 &lt;= 0.389265177607656"</span></a>
<a class="sourceLine" id="cb21-22" data-line-number="22"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb21-23" data-line-number="23"><span class="co">#&gt; $`2`$majority</span></a>
<a class="sourceLine" id="cb21-24" data-line-number="24"><span class="co">#&gt; [1] "EIF5A &gt; 2.09014605183273"</span></a>
<a class="sourceLine" id="cb21-25" data-line-number="25"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb21-26" data-line-number="26"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb21-27" data-line-number="27"><span class="co">#&gt; $`3`</span></a>
<a class="sourceLine" id="cb21-28" data-line-number="28"><span class="co">#&gt; $`3`$all</span></a>
<a class="sourceLine" id="cb21-29" data-line-number="29"><span class="co">#&gt; NULL</span></a>
<a class="sourceLine" id="cb21-30" data-line-number="30"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb21-31" data-line-number="31"><span class="co">#&gt; $`3`$majority</span></a>
<a class="sourceLine" id="cb21-32" data-line-number="32"><span class="co">#&gt; NULL</span></a></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1">gating_genes &lt;-<span class="st"> </span>validation_results<span class="op">$</span>gating_genes</a>
<a class="sourceLine" id="cb22-2" data-line-number="2">gating_genes</a>
<a class="sourceLine" id="cb22-3" data-line-number="3"><span class="co">#&gt; [1] "CD3D"     "OCIAD2"   "XIST"     "GAL"      "ARHGAP30" "EIF5A"</span></a></code></pre></div>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1">g1 &lt;-<span class="st"> </span><span class="kw"><a href="reference/plot_flowstyle.html">plot_flowstyle</a></span>(seurat_mix5050, <span class="dt">markernames =</span> gating_genes)</a>
<a class="sourceLine" id="cb23-2" data-line-number="2">g2 &lt;-<span class="st"> </span><span class="kw"><a href="reference/plot_flowstyle.html">plot_flowstyle</a></span>(seurat_mix9901, <span class="dt">markernames =</span> gating_genes)</a>
<a class="sourceLine" id="cb23-3" data-line-number="3"> </a>
<a class="sourceLine" id="cb23-4" data-line-number="4">g1</a></code></pre></div>
<p><img src="sctree_readme_full_files/figure-gfm/unnamed-chunk-21-1.png"><!-- --></p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1">g2</a></code></pre></div>
<p><img src="sctree_readme_full_files/figure-gfm/unnamed-chunk-21-2.png"><!-- --></p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1">g2[<span class="dv">1</span>,<span class="dv">2</span>]</a></code></pre></div>
<p><img src="sctree_readme_full_files/figure-gfm/unnamed-chunk-22-1.png"><!-- --></p>
</div>
<div id="finding-antibodies-for-the-experiment" class="section level2">
<h2 class="hasAnchor">
<a href="#finding-antibodies-for-the-experiment" class="anchor"></a>Finding antibodies for the experiment</h2>
<p>Since we acknowledge most experimental workflows need antibodies. We have implemented several functions to look for antibodies in vendor websites, as well as some helper functions to find the other posible aliases a gene might have.</p>
<p>Here is a simple example for a gene widely know to have an antibody available</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw"><a href="reference/query_biocompare_antibodies.html">query_biocompare_antibodies</a></span>(<span class="st">"CD11b"</span>))</a>
<a class="sourceLine" id="cb26-2" data-line-number="2"><span class="co">#&gt;                                                                       title</span></a>
<a class="sourceLine" id="cb26-3" data-line-number="3"><span class="co">#&gt; 1                                                 Anti-CD11b/ITGAM Antibody</span></a>
<a class="sourceLine" id="cb26-4" data-line-number="4"><span class="co">#&gt; 2                                        Anti-CD11b/ITGAM Picoband Antibody</span></a>
<a class="sourceLine" id="cb26-5" data-line-number="5"><span class="co">#&gt; 3 Anti-Human CD11b Monoclonal Antibody mFluor450 Conjugated, Flow Validated</span></a>
<a class="sourceLine" id="cb26-6" data-line-number="6"><span class="co">#&gt; 4     Anti-CD11b (integrin alpha-M) Rabbit Monoclonal Antibody, Clone#RM290</span></a>
<a class="sourceLine" id="cb26-7" data-line-number="7"><span class="co">#&gt; 5                                      Monoclonal Antibody to CD11b (human)</span></a>
<a class="sourceLine" id="cb26-8" data-line-number="8"><span class="co">#&gt; 6                                                      Mouse Anti-Rat CD11b</span></a>
<a class="sourceLine" id="cb26-9" data-line-number="9"><span class="co">#&gt;            vendor</span></a>
<a class="sourceLine" id="cb26-10" data-line-number="10"><span class="co">#&gt; 1       BosterBio</span></a>
<a class="sourceLine" id="cb26-11" data-line-number="11"><span class="co">#&gt; 2       BosterBio</span></a>
<a class="sourceLine" id="cb26-12" data-line-number="12"><span class="co">#&gt; 3       BosterBio</span></a>
<a class="sourceLine" id="cb26-13" data-line-number="13"><span class="co">#&gt; 4       BosterBio</span></a>
<a class="sourceLine" id="cb26-14" data-line-number="14"><span class="co">#&gt; 5 MyBioSource.com</span></a>
<a class="sourceLine" id="cb26-15" data-line-number="15"><span class="co">#&gt; 6      RayBiotech</span></a>
<a class="sourceLine" id="cb26-16" data-line-number="16"><span class="co">#&gt;                                                                                                          specification</span></a>
<a class="sourceLine" id="cb26-17" data-line-number="17"><span class="co">#&gt; 1           Applications: Western Blot (WB); Reactivity: Hu, Ms, Rt; Conjugate/Tag: Unconjugated; Quantity: 100ug/vial</span></a>
<a class="sourceLine" id="cb26-18" data-line-number="18"><span class="co">#&gt; 2 Applications: WB, FCM, ICC, IHC-fr, IHC-p; Reactivity: Hu, Ms, Rt; Conjugate/Tag: Unconjugated; Quantity: 100ug/vial</span></a>
<a class="sourceLine" id="cb26-19" data-line-number="19"><span class="co">#&gt; 3  Applications: Flow Cytometry (FCM); Reactivity: Human (Hu); Conjugate/Tag: mFluor450; Quantity: 25 Tests, 100 Tests</span></a>
<a class="sourceLine" id="cb26-20" data-line-number="20"><span class="co">#&gt; 4                          Applications: WB, IHC; Reactivity: Human (Hu); Conjugate/Tag: Unconjugated; Quantity: 100uL</span></a>
<a class="sourceLine" id="cb26-21" data-line-number="21"><span class="co">#&gt; 5            Applications: Flow Cytometry (FCM); Reactivity: Human (Hu); Conjugate/Tag: Unconjugated; Quantity: 0.1 mg</span></a>
<a class="sourceLine" id="cb26-22" data-line-number="22"><span class="co">#&gt; 6                                    Applications: Flow Cytometry (FCM); Reactivity: Rat (Rt); Quantity: 500 &lt;U+00B5&gt;g</span></a></code></pre></div>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1">aliases &lt;-<span class="st"> </span><span class="kw"><a href="reference/get_aliases.html">get_aliases</a></span>(gating_genes[[<span class="dv">1</span>]])</a>
<a class="sourceLine" id="cb27-2" data-line-number="2"></a>
<a class="sourceLine" id="cb27-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/pkg/Seurat/man/print.DimReduc.html">print</a></span>(aliases)</a>
<a class="sourceLine" id="cb27-4" data-line-number="4"><span class="co">#&gt; $CD3D</span></a>
<a class="sourceLine" id="cb27-5" data-line-number="5"><span class="co">#&gt; [1] "CD3-DELTA" "IMD19"     "T3D"       "CD3D"</span></a>
<a class="sourceLine" id="cb27-6" data-line-number="6"></a>
<a class="sourceLine" id="cb27-7" data-line-number="7"><span class="kw"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(aliases[[<span class="dv">1</span>]], <span class="cf">function</span>(alias) {</a>
<a class="sourceLine" id="cb27-8" data-line-number="8">     <span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw"><a href="reference/query_biocompare_antibodies.html">query_biocompare_antibodies</a></span>(alias))</a>
<a class="sourceLine" id="cb27-9" data-line-number="9">})</a>
<a class="sourceLine" id="cb27-10" data-line-number="10"><span class="co">#&gt; [[1]]</span></a>
<a class="sourceLine" id="cb27-11" data-line-number="11"><span class="co">#&gt;                                                                title</span></a>
<a class="sourceLine" id="cb27-12" data-line-number="12"><span class="co">#&gt; 1  Anti-Human CD3 Monoclonal Antibody APC Conjugated, Flow Validated</span></a>
<a class="sourceLine" id="cb27-13" data-line-number="13"><span class="co">#&gt; 2 Anti-Human CD3 Monoclonal Antibody FITC Conjugated, Flow Validated</span></a>
<a class="sourceLine" id="cb27-14" data-line-number="14"><span class="co">#&gt; 3   Anti-Human CD3 Monoclonal Antibody PE Conjugated, Flow Validated</span></a>
<a class="sourceLine" id="cb27-15" data-line-number="15"><span class="co">#&gt; 4                               CD3d / CD3 delta Antibody, Mouse MAb</span></a>
<a class="sourceLine" id="cb27-16" data-line-number="16"><span class="co">#&gt; 5                                                 CD3 delta antibody</span></a>
<a class="sourceLine" id="cb27-17" data-line-number="17"><span class="co">#&gt; 6                                      CD3-delta Polyclonal Antibody</span></a>
<a class="sourceLine" id="cb27-18" data-line-number="18"><span class="co">#&gt;                  vendor</span></a>
<a class="sourceLine" id="cb27-19" data-line-number="19"><span class="co">#&gt; 1             BosterBio</span></a>
<a class="sourceLine" id="cb27-20" data-line-number="20"><span class="co">#&gt; 2             BosterBio</span></a>
<a class="sourceLine" id="cb27-21" data-line-number="21"><span class="co">#&gt; 3             BosterBio</span></a>
<a class="sourceLine" id="cb27-22" data-line-number="22"><span class="co">#&gt; 4 Sino Biological, Inc.</span></a>
<a class="sourceLine" id="cb27-23" data-line-number="23"><span class="co">#&gt; 5               Biorbyt</span></a>
<a class="sourceLine" id="cb27-24" data-line-number="24"><span class="co">#&gt; 6       MyBioSource.com</span></a>
<a class="sourceLine" id="cb27-25" data-line-number="25"><span class="co">#&gt;                                                                                                                      specification</span></a>
<a class="sourceLine" id="cb27-26" data-line-number="26"><span class="co">#&gt; 1         Applications: Flow Cytometry (FCM); Reactivity: Human (Hu); Conjugate/Tag: APC; Quantity: 25 tests, 100 tests, 500 tests</span></a>
<a class="sourceLine" id="cb27-27" data-line-number="27"><span class="co">#&gt; 2                   Applications: Flow Cytometry (FCM); Reactivity: Human (Hu); Conjugate/Tag: FITC; Quantity: 25 tests, 100 tests</span></a>
<a class="sourceLine" id="cb27-28" data-line-number="28"><span class="co">#&gt; 3          Applications: Flow Cytometry (FCM); Reactivity: Human (Hu); Conjugate/Tag: PE; Quantity: 25 tests, 100 tests, 500 tests</span></a>
<a class="sourceLine" id="cb27-29" data-line-number="29"><span class="co">#&gt; 4 Applications: Immunohistochemistry-paraffin (IHC-p); Reactivity: Human (Hu); Conjugate/Tag: Unconjugated; Quantity: 100&lt;U+00B5&gt;L</span></a>
<a class="sourceLine" id="cb27-30" data-line-number="30"><span class="co">#&gt; 5                                   Applications: WB, ELISA; Reactivity: Human (Hu); Conjugate/Tag: Unconjugated; Quantity: 100 ug</span></a>
<a class="sourceLine" id="cb27-31" data-line-number="31"><span class="co">#&gt; 6                                  Applications: WB, ELISA; Reactivity: Human (Hu); Conjugate/Tag: Unconjugated; Quantity: 0.05 ml</span></a>
<a class="sourceLine" id="cb27-32" data-line-number="32"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb27-33" data-line-number="33"><span class="co">#&gt; [[2]]</span></a>
<a class="sourceLine" id="cb27-34" data-line-number="34"><span class="co">#&gt;                                                                title</span></a>
<a class="sourceLine" id="cb27-35" data-line-number="35"><span class="co">#&gt; 1                               Anti-CD3D Rabbit Monoclonal Antibody</span></a>
<a class="sourceLine" id="cb27-36" data-line-number="36"><span class="co">#&gt; 2  Anti-Human CD3 Monoclonal Antibody APC Conjugated, Flow Validated</span></a>
<a class="sourceLine" id="cb27-37" data-line-number="37"><span class="co">#&gt; 3 Anti-Human CD3 Monoclonal Antibody FITC Conjugated, Flow Validated</span></a>
<a class="sourceLine" id="cb27-38" data-line-number="38"><span class="co">#&gt; 4   Anti-Human CD3 Monoclonal Antibody PE Conjugated, Flow Validated</span></a>
<a class="sourceLine" id="cb27-39" data-line-number="39"><span class="co">#&gt; 5                                           CD3D Monoclonal Antibody</span></a>
<a class="sourceLine" id="cb27-40" data-line-number="40"><span class="co">#&gt; 6                                           CD3D Monoclonal Antibody</span></a>
<a class="sourceLine" id="cb27-41" data-line-number="41"><span class="co">#&gt;                 vendor</span></a>
<a class="sourceLine" id="cb27-42" data-line-number="42"><span class="co">#&gt; 1            BosterBio</span></a>
<a class="sourceLine" id="cb27-43" data-line-number="43"><span class="co">#&gt; 2            BosterBio</span></a>
<a class="sourceLine" id="cb27-44" data-line-number="44"><span class="co">#&gt; 3            BosterBio</span></a>
<a class="sourceLine" id="cb27-45" data-line-number="45"><span class="co">#&gt; 4            BosterBio</span></a>
<a class="sourceLine" id="cb27-46" data-line-number="46"><span class="co">#&gt; 5 LifeSpan BioSciences</span></a>
<a class="sourceLine" id="cb27-47" data-line-number="47"><span class="co">#&gt; 6 LifeSpan BioSciences</span></a>
<a class="sourceLine" id="cb27-48" data-line-number="48"><span class="co">#&gt;                                                                                                              specification</span></a>
<a class="sourceLine" id="cb27-49" data-line-number="49"><span class="co">#&gt; 1                      Applications: WB, IF, IP; Reactivity: Human (Hu); Conjugate/Tag: Unconjugated; Quantity: 100ug/vial</span></a>
<a class="sourceLine" id="cb27-50" data-line-number="50"><span class="co">#&gt; 2 Applications: Flow Cytometry (FCM); Reactivity: Human (Hu); Conjugate/Tag: APC; Quantity: 25 tests, 100 tests, 500 tests</span></a>
<a class="sourceLine" id="cb27-51" data-line-number="51"><span class="co">#&gt; 3           Applications: Flow Cytometry (FCM); Reactivity: Human (Hu); Conjugate/Tag: FITC; Quantity: 25 tests, 100 tests</span></a>
<a class="sourceLine" id="cb27-52" data-line-number="52"><span class="co">#&gt; 4  Applications: Flow Cytometry (FCM); Reactivity: Human (Hu); Conjugate/Tag: PE; Quantity: 25 tests, 100 tests, 500 tests</span></a>
<a class="sourceLine" id="cb27-53" data-line-number="53"><span class="co">#&gt; 5                         Applications: ELISA; Reactivity: Human (Hu); Conjugate/Tag: Unconjugated; Quantity: 50 &lt;U+00B5&gt;l</span></a>
<a class="sourceLine" id="cb27-54" data-line-number="54"><span class="co">#&gt; 6                    Applications: WB, ELISA; Reactivity: Human (Hu); Conjugate/Tag: Unconjugated; Quantity: 100 &lt;U+00B5&gt;g</span></a>
<a class="sourceLine" id="cb27-55" data-line-number="55"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb27-56" data-line-number="56"><span class="co">#&gt; [[3]]</span></a>
<a class="sourceLine" id="cb27-57" data-line-number="57"><span class="co">#&gt;                                                                title</span></a>
<a class="sourceLine" id="cb27-58" data-line-number="58"><span class="co">#&gt; 1                               Anti-CD3D Rabbit Monoclonal Antibody</span></a>
<a class="sourceLine" id="cb27-59" data-line-number="59"><span class="co">#&gt; 2  Anti-Human CD3 Monoclonal Antibody APC Conjugated, Flow Validated</span></a>
<a class="sourceLine" id="cb27-60" data-line-number="60"><span class="co">#&gt; 3 Anti-Human CD3 Monoclonal Antibody FITC Conjugated, Flow Validated</span></a>
<a class="sourceLine" id="cb27-61" data-line-number="61"><span class="co">#&gt; 4   Anti-Human CD3 Monoclonal Antibody PE Conjugated, Flow Validated</span></a>
<a class="sourceLine" id="cb27-62" data-line-number="62"><span class="co">#&gt; 5                                                 anti-CD3D Antibody</span></a>
<a class="sourceLine" id="cb27-63" data-line-number="63"><span class="co">#&gt; 6                                                 anti-CD3D Antibody</span></a>
<a class="sourceLine" id="cb27-64" data-line-number="64"><span class="co">#&gt;              vendor</span></a>
<a class="sourceLine" id="cb27-65" data-line-number="65"><span class="co">#&gt; 1         BosterBio</span></a>
<a class="sourceLine" id="cb27-66" data-line-number="66"><span class="co">#&gt; 2         BosterBio</span></a>
<a class="sourceLine" id="cb27-67" data-line-number="67"><span class="co">#&gt; 3         BosterBio</span></a>
<a class="sourceLine" id="cb27-68" data-line-number="68"><span class="co">#&gt; 4         BosterBio</span></a>
<a class="sourceLine" id="cb27-69" data-line-number="69"><span class="co">#&gt; 5 antibodies-online</span></a>
<a class="sourceLine" id="cb27-70" data-line-number="70"><span class="co">#&gt; 6 antibodies-online</span></a>
<a class="sourceLine" id="cb27-71" data-line-number="71"><span class="co">#&gt;                                                                                                              specification</span></a>
<a class="sourceLine" id="cb27-72" data-line-number="72"><span class="co">#&gt; 1                      Applications: WB, IF, IP; Reactivity: Human (Hu); Conjugate/Tag: Unconjugated; Quantity: 100ug/vial</span></a>
<a class="sourceLine" id="cb27-73" data-line-number="73"><span class="co">#&gt; 2 Applications: Flow Cytometry (FCM); Reactivity: Human (Hu); Conjugate/Tag: APC; Quantity: 25 tests, 100 tests, 500 tests</span></a>
<a class="sourceLine" id="cb27-74" data-line-number="74"><span class="co">#&gt; 3           Applications: Flow Cytometry (FCM); Reactivity: Human (Hu); Conjugate/Tag: FITC; Quantity: 25 tests, 100 tests</span></a>
<a class="sourceLine" id="cb27-75" data-line-number="75"><span class="co">#&gt; 4  Applications: Flow Cytometry (FCM); Reactivity: Human (Hu); Conjugate/Tag: PE; Quantity: 25 tests, 100 tests, 500 tests</span></a>
<a class="sourceLine" id="cb27-76" data-line-number="76"><span class="co">#&gt; 5                  Applications: WB, IHC; Reactivity: Hu, Ms, Rt, Mu; Conjugate/Tag: Unconjugated; Quantity: 200 &lt;U+00B5&gt;L</span></a>
<a class="sourceLine" id="cb27-77" data-line-number="77"><span class="co">#&gt; 6                   Applications: WB, IHC; Reactivity: Hu, Ms, Rt, Mu; Conjugate/Tag: Unconjugated; Quantity: 50 &lt;U+00B5&gt;L</span></a>
<a class="sourceLine" id="cb27-78" data-line-number="78"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb27-79" data-line-number="79"><span class="co">#&gt; [[4]]</span></a>
<a class="sourceLine" id="cb27-80" data-line-number="80"><span class="co">#&gt;                                  title                vendor</span></a>
<a class="sourceLine" id="cb27-81" data-line-number="81"><span class="co">#&gt; 1 Anti-CD3D Rabbit Monoclonal Antibody             BosterBio</span></a>
<a class="sourceLine" id="cb27-82" data-line-number="82"><span class="co">#&gt; 2     CD3D &amp; CD3E Antibody, Rabbit PAb Sino Biological, Inc.</span></a>
<a class="sourceLine" id="cb27-83" data-line-number="83"><span class="co">#&gt; 3 CD3d / CD3 delta Antibody, Mouse MAb Sino Biological, Inc.</span></a>
<a class="sourceLine" id="cb27-84" data-line-number="84"><span class="co">#&gt; 4 CD3d / CD3 delta Antibody, Mouse MAb Sino Biological, Inc.</span></a>
<a class="sourceLine" id="cb27-85" data-line-number="85"><span class="co">#&gt; 5             CD3D Monoclonal Antibody  LifeSpan BioSciences</span></a>
<a class="sourceLine" id="cb27-86" data-line-number="86"><span class="co">#&gt; 6             CD3D Monoclonal Antibody  LifeSpan BioSciences</span></a>
<a class="sourceLine" id="cb27-87" data-line-number="87"><span class="co">#&gt;                                                                                                                      specification</span></a>
<a class="sourceLine" id="cb27-88" data-line-number="88"><span class="co">#&gt; 1                              Applications: WB, IF, IP; Reactivity: Human (Hu); Conjugate/Tag: Unconjugated; Quantity: 100ug/vial</span></a>
<a class="sourceLine" id="cb27-89" data-line-number="89"><span class="co">#&gt; 2                                 Applications: ELISA; Reactivity: Human (Hu); Conjugate/Tag: Unconjugated; Quantity: 100&lt;U+00B5&gt;L</span></a>
<a class="sourceLine" id="cb27-90" data-line-number="90"><span class="co">#&gt; 3 Applications: Immunohistochemistry-paraffin (IHC-p); Reactivity: Human (Hu); Conjugate/Tag: Unconjugated; Quantity: 100&lt;U+00B5&gt;L</span></a>
<a class="sourceLine" id="cb27-91" data-line-number="91"><span class="co">#&gt; 4                                 Applications: ELISA; Reactivity: Human (Hu); Conjugate/Tag: Unconjugated; Quantity: 100&lt;U+00B5&gt;L</span></a>
<a class="sourceLine" id="cb27-92" data-line-number="92"><span class="co">#&gt; 5                                 Applications: ELISA; Reactivity: Human (Hu); Conjugate/Tag: Unconjugated; Quantity: 50 &lt;U+00B5&gt;l</span></a>
<a class="sourceLine" id="cb27-93" data-line-number="93"><span class="co">#&gt; 6                            Applications: WB, ELISA; Reactivity: Human (Hu); Conjugate/Tag: Unconjugated; Quantity: 100 &lt;U+00B5&gt;g</span></a></code></pre></div>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()</a>
<a class="sourceLine" id="cb28-2" data-line-number="2"><span class="co">#&gt; R version 3.5.2 (2018-12-20)</span></a>
<a class="sourceLine" id="cb28-3" data-line-number="3"><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></a>
<a class="sourceLine" id="cb28-4" data-line-number="4"><span class="co">#&gt; Running under: CentOS Linux 7 (Core)</span></a>
<a class="sourceLine" id="cb28-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb28-6" data-line-number="6"><span class="co">#&gt; Matrix products: default</span></a>
<a class="sourceLine" id="cb28-7" data-line-number="7"><span class="co">#&gt; BLAS: /home/jpaezpae/opt/R-3.5.2/lib/libRblas.so</span></a>
<a class="sourceLine" id="cb28-8" data-line-number="8"><span class="co">#&gt; LAPACK: /home/jpaezpae/opt/R-3.5.2/lib/libRlapack.so</span></a>
<a class="sourceLine" id="cb28-9" data-line-number="9"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb28-10" data-line-number="10"><span class="co">#&gt; locale:</span></a>
<a class="sourceLine" id="cb28-11" data-line-number="11"><span class="co">#&gt; [1] C</span></a>
<a class="sourceLine" id="cb28-12" data-line-number="12"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb28-13" data-line-number="13"><span class="co">#&gt; attached base packages:</span></a>
<a class="sourceLine" id="cb28-14" data-line-number="14"><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></a>
<a class="sourceLine" id="cb28-15" data-line-number="15"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb28-16" data-line-number="16"><span class="co">#&gt; other attached packages:</span></a>
<a class="sourceLine" id="cb28-17" data-line-number="17"><span class="co">#&gt; [1] Seurat_2.3.4      Matrix_1.2-15     cowplot_0.9.4     ggplot2_3.1.0    </span></a>
<a class="sourceLine" id="cb28-18" data-line-number="18"><span class="co">#&gt; [5] sctree_0.0.1.9002</span></a>
<a class="sourceLine" id="cb28-19" data-line-number="19"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb28-20" data-line-number="20"><span class="co">#&gt; loaded via a namespace (and not attached):</span></a>
<a class="sourceLine" id="cb28-21" data-line-number="21"><span class="co">#&gt;   [1] snow_0.4-3           backports_1.1.3      Hmisc_4.2-0         </span></a>
<a class="sourceLine" id="cb28-22" data-line-number="22"><span class="co">#&gt;   [4] selectr_0.4-1        wrapr_1.8.6          plyr_1.8.4          </span></a>
<a class="sourceLine" id="cb28-23" data-line-number="23"><span class="co">#&gt;   [7] igraph_1.2.4         lazyeval_0.2.2       splines_3.5.2       </span></a>
<a class="sourceLine" id="cb28-24" data-line-number="24"><span class="co">#&gt;  [10] digest_0.6.18        foreach_1.4.4        htmltools_0.3.6     </span></a>
<a class="sourceLine" id="cb28-25" data-line-number="25"><span class="co">#&gt;  [13] viridis_0.5.1        lars_1.2             gdata_2.18.0        </span></a>
<a class="sourceLine" id="cb28-26" data-line-number="26"><span class="co">#&gt;  [16] magrittr_1.5         checkmate_1.9.1      memoise_1.1.0       </span></a>
<a class="sourceLine" id="cb28-27" data-line-number="27"><span class="co">#&gt;  [19] cluster_2.0.7-1      mixtools_1.1.0       ROCR_1.0-7          </span></a>
<a class="sourceLine" id="cb28-28" data-line-number="28"><span class="co">#&gt;  [22] R.utils_2.8.0        colorspace_1.4-1     blob_1.1.1          </span></a>
<a class="sourceLine" id="cb28-29" data-line-number="29"><span class="co">#&gt;  [25] rvest_0.3.2          xfun_0.6             dplyr_0.8.0.1       </span></a>
<a class="sourceLine" id="cb28-30" data-line-number="30"><span class="co">#&gt;  [28] crayon_1.3.4         jsonlite_1.6         libcoin_1.0-4       </span></a>
<a class="sourceLine" id="cb28-31" data-line-number="31"><span class="co">#&gt;  [31] survival_2.43-3      zoo_1.8-5            iterators_1.0.10    </span></a>
<a class="sourceLine" id="cb28-32" data-line-number="32"><span class="co">#&gt;  [34] ape_5.3              glue_1.3.1           gtable_0.3.0        </span></a>
<a class="sourceLine" id="cb28-33" data-line-number="33"><span class="co">#&gt;  [37] kernlab_0.9-27       prabclus_2.2-7       BiocGenerics_0.26.0 </span></a>
<a class="sourceLine" id="cb28-34" data-line-number="34"><span class="co">#&gt;  [40] DEoptimR_1.0-8       scales_1.0.0         mvtnorm_1.0-10      </span></a>
<a class="sourceLine" id="cb28-35" data-line-number="35"><span class="co">#&gt;  [43] DBI_1.0.0            GGally_1.4.0         bibtex_0.4.2        </span></a>
<a class="sourceLine" id="cb28-36" data-line-number="36"><span class="co">#&gt;  [46] Rcpp_1.0.1           metap_1.1            dtw_1.20-1          </span></a>
<a class="sourceLine" id="cb28-37" data-line-number="37"><span class="co">#&gt;  [49] viridisLite_0.3.0    xtable_1.8-3         htmlTable_1.13.1    </span></a>
<a class="sourceLine" id="cb28-38" data-line-number="38"><span class="co">#&gt;  [52] reticulate_1.11.1    foreign_0.8-71       bit_1.1-14          </span></a>
<a class="sourceLine" id="cb28-39" data-line-number="39"><span class="co">#&gt;  [55] proxy_0.4-23         mclust_5.4.3         SDMTools_1.1-221    </span></a>
<a class="sourceLine" id="cb28-40" data-line-number="40"><span class="co">#&gt;  [58] Formula_1.2-3        stats4_3.5.2         tsne_0.1-3          </span></a>
<a class="sourceLine" id="cb28-41" data-line-number="41"><span class="co">#&gt;  [61] DT_0.5               htmlwidgets_1.3      httr_1.4.0          </span></a>
<a class="sourceLine" id="cb28-42" data-line-number="42"><span class="co">#&gt;  [64] gplots_3.0.1.1       RColorBrewer_1.1-2   fpc_2.1-11.1        </span></a>
<a class="sourceLine" id="cb28-43" data-line-number="43"><span class="co">#&gt;  [67] acepack_1.4.1        modeltools_0.2-22    ica_1.0-2           </span></a>
<a class="sourceLine" id="cb28-44" data-line-number="44"><span class="co">#&gt;  [70] pkgconfig_2.0.2      reshape_0.8.8        R.methodsS3_1.7.1   </span></a>
<a class="sourceLine" id="cb28-45" data-line-number="45"><span class="co">#&gt;  [73] flexmix_2.3-15       nnet_7.3-12          labeling_0.3        </span></a>
<a class="sourceLine" id="cb28-46" data-line-number="46"><span class="co">#&gt;  [76] tidyselect_0.2.5     rlang_0.3.3          reshape2_1.4.3      </span></a>
<a class="sourceLine" id="cb28-47" data-line-number="47"><span class="co">#&gt;  [79] later_0.8.0          AnnotationDbi_1.42.1 munsell_0.5.0       </span></a>
<a class="sourceLine" id="cb28-48" data-line-number="48"><span class="co">#&gt;  [82] tools_3.5.2          RSQLite_2.1.1        ranger_0.11.2       </span></a>
<a class="sourceLine" id="cb28-49" data-line-number="49"><span class="co">#&gt;  [85] ggridges_0.5.1       evaluate_0.13        stringr_1.4.0       </span></a>
<a class="sourceLine" id="cb28-50" data-line-number="50"><span class="co">#&gt;  [88] yaml_2.2.0           npsurv_0.4-0         org.Hs.eg.db_3.4.0  </span></a>
<a class="sourceLine" id="cb28-51" data-line-number="51"><span class="co">#&gt;  [91] knitr_1.22           bit64_0.9-7          fitdistrplus_1.0-14 </span></a>
<a class="sourceLine" id="cb28-52" data-line-number="52"><span class="co">#&gt;  [94] robustbase_0.93-4    caTools_1.17.1.2     purrr_0.3.2         </span></a>
<a class="sourceLine" id="cb28-53" data-line-number="53"><span class="co">#&gt;  [97] RANN_2.6.1           pbapply_1.4-0        nlme_3.1-137        </span></a>
<a class="sourceLine" id="cb28-54" data-line-number="54"><span class="co">#&gt; [100] mime_0.6             R.oo_1.22.0          xml2_1.2.0          </span></a>
<a class="sourceLine" id="cb28-55" data-line-number="55"><span class="co">#&gt; [103] hdf5r_1.1.1          compiler_3.5.2       rstudioapi_0.10     </span></a>
<a class="sourceLine" id="cb28-56" data-line-number="56"><span class="co">#&gt; [106] curl_3.3             png_0.1-7            lsei_1.2-0          </span></a>
<a class="sourceLine" id="cb28-57" data-line-number="57"><span class="co">#&gt; [109] tibble_2.1.1         stringi_1.4.3        lattice_0.20-38     </span></a>
<a class="sourceLine" id="cb28-58" data-line-number="58"><span class="co">#&gt; [112] trimcluster_0.1-2.1  pillar_1.3.1         Rdpack_0.10-1       </span></a>
<a class="sourceLine" id="cb28-59" data-line-number="59"><span class="co">#&gt; [115] lmtest_0.9-36        data.table_1.12.0    bitops_1.0-6        </span></a>
<a class="sourceLine" id="cb28-60" data-line-number="60"><span class="co">#&gt; [118] irlba_2.3.3          gbRd_0.4-11          httpuv_1.5.1        </span></a>
<a class="sourceLine" id="cb28-61" data-line-number="61"><span class="co">#&gt; [121] R6_2.4.0             latticeExtra_0.6-28  promises_1.0.1      </span></a>
<a class="sourceLine" id="cb28-62" data-line-number="62"><span class="co">#&gt; [124] KernSmooth_2.23-15   gridExtra_2.3        IRanges_2.14.12     </span></a>
<a class="sourceLine" id="cb28-63" data-line-number="63"><span class="co">#&gt; [127] codetools_0.2-16     MASS_7.3-51.1        gtools_3.8.1        </span></a>
<a class="sourceLine" id="cb28-64" data-line-number="64"><span class="co">#&gt; [130] assertthat_0.2.1     withr_2.1.2          S4Vectors_0.18.3    </span></a>
<a class="sourceLine" id="cb28-65" data-line-number="65"><span class="co">#&gt; [133] diptest_0.75-7       parallel_3.5.2       doSNOW_1.0.16       </span></a>
<a class="sourceLine" id="cb28-66" data-line-number="66"><span class="co">#&gt; [136] grid_3.5.2           rpart_4.1-13         tidyr_0.8.3         </span></a>
<a class="sourceLine" id="cb28-67" data-line-number="67"><span class="co">#&gt; [139] class_7.3-14         rmarkdown_1.11       inum_1.0-0          </span></a>
<a class="sourceLine" id="cb28-68" data-line-number="68"><span class="co">#&gt; [142] segmented_0.5-3.0    Rtsne_0.15           partykit_1.2-3      </span></a>
<a class="sourceLine" id="cb28-69" data-line-number="69"><span class="co">#&gt; [145] Biobase_2.40.0       shiny_1.2.0          base64enc_0.1-3</span></a></code></pre></div>
</div>
</div>
<div id="reproducing-the-runs-in-the-purdue-cluster" class="section level1">
<h1 class="hasAnchor">
<a href="#reproducing-the-runs-in-the-purdue-cluster" class="anchor"></a>Reproducing the runs in the purdue cluster</h1>
<p>To reproduce the runs in the purdue cluster run as follows …</p>
<ol>
<li>We get the data from the temporary directory</li>
</ol>
<!-- end list --><div class="sourceCode" id="cb29"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1">git clone https<span class="op">:</span><span class="er">//</span>github.rcac.purdue.edu<span class="op">/</span>jpaezpae<span class="op">/</span>data_sctree data</a>
<a class="sourceLine" id="cb29-2" data-line-number="2">cd data</a>
<a class="sourceLine" id="cb29-3" data-line-number="3">bash untar_data.bash</a></code></pre></div>
<ol>
<li>We run the standard seurat workflow.</li>
</ol>
<p>This will output a report and generate an .RDS file for each of the final seurat objects</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" data-line-number="1">bash .<span class="op">/</span>bash<span class="op">/</span>build_jobs_seurat_workflow.bash .<span class="op">/</span>data<span class="op">/</span>filtered_matrices_mex_<span class="dv">5050</span><span class="op">/</span>hg19<span class="op">/</span><span class="st"> </span>mix5050</a>
<a class="sourceLine" id="cb30-2" data-line-number="2">bash .<span class="op">/</span>bash<span class="op">/</span>build_jobs_seurat_workflow.bash .<span class="op">/</span>data<span class="op">/</span>filtered_matrices_mex_<span class="dv">9901</span><span class="op">/</span>hg19<span class="op">/</span><span class="st"> </span>mix9901 </a></code></pre></div>
<ol>
<li>Whenever those are done, run this …</li>
</ol>
<p>This will run the benchmarks for the datasets. Will also generate 2 .RDS files containing a list with a lot of stuff in it.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1"><span class="cf">for</span> i <span class="cf">in</span> seurat<span class="op">*</span>.RDS ; do bash .<span class="op">/</span>bash<span class="op">/</span>build_jobs_acc_benchmark.bash <span class="op">$</span>i ; done</a></code></pre></div>
</div>
<div id="steps-down-the-road" class="section level1">
<h1 class="hasAnchor">
<a href="#steps-down-the-road" class="anchor"></a>Steps down the road</h1>
<ol>
<li>Make figure list.</li>
<li>Address some of the TODO’s in this repository</li>
<li>Reduce dependecies by replacing functions to base equivalents.</li>
<li>Add links to the documentation to make nicer to explore the package from inside R</li>
<li>Implement plot that actually illustrates the progressive gating in the decision tree</li>
<li>DRASTICALLY increase code coverage (right now everything runs without errors due to R CMD check) but expectations are not tested</li>
<li>Implement a way to find markers for clusters exclusively upregulated</li>
</ol>
</div>


  </div>

</div>



      <footer>
      <div class="copyright">
  <p>Developed by J. Sebastian Paez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


