<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Gene expression markers of identity classes — FindMarkers • sctree</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Gene expression markers of identity classes — FindMarkers" />
<meta property="og:description" content="Finds markers (differentially expressed genes) for identity classes" />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sctree</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.6.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/sctree.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/antibody_query.html">Query Antibody vendors from inside R</a>
    </li>
    <li>
      <a href="../articles/plot_flowstyle.html">Flow style plotting</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Gene expression markers of identity classes</h1>
    
    <div class="hidden name"><code>FindMarkers.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Finds markers (differentially expressed genes) for identity classes</p>
    </div>

    <pre class="usage"><span class='fu'>FindMarkers</span>(<span class='no'>object</span>, <span class='no'>...</span>)

<span class='co'># S3 method for default</span>
<span class='fu'>FindMarkers</span>(
  <span class='no'>object</span>,
  <span class='kw'>slot</span> <span class='kw'>=</span> <span class='st'>"data"</span>,
  <span class='kw'>counts</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/numeric.html'>numeric</a></span>(),
  <span class='kw'>cells.1</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>cells.2</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>features</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>reduction</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>logfc.threshold</span> <span class='kw'>=</span> <span class='fl'>0.25</span>,
  <span class='kw'>test.use</span> <span class='kw'>=</span> <span class='st'>"wilcox"</span>,
  <span class='kw'>min.pct</span> <span class='kw'>=</span> <span class='fl'>0.1</span>,
  <span class='kw'>min.diff.pct</span> <span class='kw'>=</span> -<span class='fl'>Inf</span>,
  <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>only.pos</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>max.cells.per.ident</span> <span class='kw'>=</span> <span class='fl'>Inf</span>,
  <span class='kw'>random.seed</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>latent.vars</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>min.cells.feature</span> <span class='kw'>=</span> <span class='fl'>3</span>,
  <span class='kw'>min.cells.group</span> <span class='kw'>=</span> <span class='fl'>3</span>,
  <span class='kw'>pseudocount.use</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='no'>...</span>
)

<span class='co'># S3 method for Seurat</span>
<span class='fu'>FindMarkers</span>(
  <span class='no'>object</span>,
  <span class='kw'>ident.1</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>ident.2</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>group.by</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>subset.ident</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>assay</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>slot</span> <span class='kw'>=</span> <span class='st'>"data"</span>,
  <span class='kw'>reduction</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>features</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>logfc.threshold</span> <span class='kw'>=</span> <span class='fl'>0.25</span>,
  <span class='kw'>test.use</span> <span class='kw'>=</span> <span class='st'>"wilcox"</span>,
  <span class='kw'>min.pct</span> <span class='kw'>=</span> <span class='fl'>0.1</span>,
  <span class='kw'>min.diff.pct</span> <span class='kw'>=</span> -<span class='fl'>Inf</span>,
  <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>only.pos</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>max.cells.per.ident</span> <span class='kw'>=</span> <span class='fl'>Inf</span>,
  <span class='kw'>random.seed</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>latent.vars</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>min.cells.feature</span> <span class='kw'>=</span> <span class='fl'>3</span>,
  <span class='kw'>min.cells.group</span> <span class='kw'>=</span> <span class='fl'>3</span>,
  <span class='kw'>pseudocount.use</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='no'>...</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>object</th>
      <td><p>An object</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Arguments passed to other methods and to specific DE methods</p></td>
    </tr>
    <tr>
      <th>slot</th>
      <td><p>Slot to pull data from; note that if <code>test.use</code> is "negbinom", "poisson", or "DESeq2",
<code>slot</code> will be set to "counts"</p></td>
    </tr>
    <tr>
      <th>counts</th>
      <td><p>Count matrix if using scale.data for DE tests. This is used for
computing pct.1 and pct.2 and for filtering features based on fraction
expressing</p></td>
    </tr>
    <tr>
      <th>cells.1</th>
      <td><p>Vector of cell names belonging to group 1</p></td>
    </tr>
    <tr>
      <th>cells.2</th>
      <td><p>Vector of cell names belonging to group 2</p></td>
    </tr>
    <tr>
      <th>features</th>
      <td><p>Genes to test. Default is to use all genes</p></td>
    </tr>
    <tr>
      <th>reduction</th>
      <td><p>Reduction to use in differential expression testing - will test for DE on cell embeddings</p></td>
    </tr>
    <tr>
      <th>logfc.threshold</th>
      <td><p>Limit testing to genes which show, on average, at least
X-fold difference (log-scale) between the two groups of cells. Default is 0.25
Increasing logfc.threshold speeds up the function, but can miss weaker signals.</p></td>
    </tr>
    <tr>
      <th>test.use</th>
      <td><p>Denotes which test to use. Available options are:</p><ul>
<li><p>"wilcox" : Identifies differentially expressed genes between two
 groups of cells using a Wilcoxon Rank Sum test (default)</p></li>
<li><p>"bimod" : Likelihood-ratio test for single cell gene expression,
 (McDavid et al., Bioinformatics, 2013)</p></li>
<li><p>"roc" : Identifies 'markers' of gene expression using ROC analysis.
 For each gene, evaluates (using AUC) a classifier built on that gene alone,
 to classify between two groups of cells. An AUC value of 1 means that
 expression values for this gene alone can perfectly classify the two
 groupings (i.e. Each of the cells in cells.1 exhibit a higher level than
 each of the cells in cells.2). An AUC value of 0 also means there is perfect
 classification, but in the other direction. A value of 0.5 implies that
 the gene has no predictive power to classify the two groups. Returns a
 'predictive power' (abs(AUC-0.5) * 2) ranked matrix of putative differentially
 expressed genes.</p></li>
<li><p>"t" : Identify differentially expressed genes between two groups of
 cells using the Student's t-test.</p></li>
<li><p>"negbinom" : Identifies differentially expressed genes between two
  groups of cells using a negative binomial generalized linear model.
  Use only for UMI-based datasets</p></li>
<li><p>"poisson" : Identifies differentially expressed genes between two
  groups of cells using a poisson generalized linear model.
  Use only for UMI-based datasets</p></li>
<li><p>"LR" : Uses a logistic regression framework to determine differentially
 expressed genes. Constructs a logistic regression model predicting group
 membership based on each feature individually and compares this to a null
 model with a likelihood ratio test.</p></li>
<li><p>"MAST" : Identifies differentially expressed genes between two groups
 of cells using a hurdle model tailored to scRNA-seq data. Utilizes the MAST
 package to run the DE testing.</p></li>
<li><p>"DESeq2" : Identifies differentially expressed genes between two groups
 of cells based on a model using DESeq2 which uses a negative binomial
 distribution (Love et al, Genome Biology, 2014).This test does not support
 pre-filtering of genes based on average difference (or percent detection rate)
 between cell groups. However, genes may be pre-filtered based on their
 minimum detection rate (min.pct) across both cell groups. To use this method,
 please install DESeq2, using the instructions at
 https://bioconductor.org/packages/release/bioc/html/DESeq2.html</p></li>
</ul></td>
    </tr>
    <tr>
      <th>min.pct</th>
      <td><p>only test genes that are detected in a minimum fraction of
min.pct cells in either of the two populations. Meant to speed up the function
by not testing genes that are very infrequently expressed. Default is 0.1</p></td>
    </tr>
    <tr>
      <th>min.diff.pct</th>
      <td><p>only test genes that show a minimum difference in the
fraction of detection between the two groups. Set to -Inf by default</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Print a progress bar once expression testing begins</p></td>
    </tr>
    <tr>
      <th>only.pos</th>
      <td><p>Only return positive markers (FALSE by default)</p></td>
    </tr>
    <tr>
      <th>max.cells.per.ident</th>
      <td><p>Down sample each identity class to a max number.
Default is no downsampling. Not activated by default (set to Inf)</p></td>
    </tr>
    <tr>
      <th>random.seed</th>
      <td><p>Random seed for downsampling</p></td>
    </tr>
    <tr>
      <th>latent.vars</th>
      <td><p>Variables to test, used only when <code>test.use</code> is one of
'LR', 'negbinom', 'poisson', or 'MAST'</p></td>
    </tr>
    <tr>
      <th>min.cells.feature</th>
      <td><p>Minimum number of cells expressing the feature in at least one
of the two groups, currently only used for poisson and negative binomial tests</p></td>
    </tr>
    <tr>
      <th>min.cells.group</th>
      <td><p>Minimum number of cells in one of the groups</p></td>
    </tr>
    <tr>
      <th>pseudocount.use</th>
      <td><p>Pseudocount to add to averaged expression values when
calculating logFC. 1 by default.</p></td>
    </tr>
    <tr>
      <th>ident.1</th>
      <td><p>Identity class to define markers for; pass an object of class
<code>phylo</code> or 'clustertree' to find markers for a node in a cluster tree;
passing 'clustertree' requires <code>BuildClusterTree</code> to have been run</p></td>
    </tr>
    <tr>
      <th>ident.2</th>
      <td><p>A second identity class for comparison; if <code>NULL</code>,
use all other cells for comparison; if an object of class <code>phylo</code> or
'clustertree' is passed to <code>ident.1</code>, must pass a node to find markers for</p></td>
    </tr>
    <tr>
      <th>group.by</th>
      <td><p>Regroup cells into a different identity class prior to performing differential expression (see example)</p></td>
    </tr>
    <tr>
      <th>subset.ident</th>
      <td><p>Subset a particular identity class prior to regrouping. Only relevant if group.by is set (see example)</p></td>
    </tr>
    <tr>
      <th>assay</th>
      <td><p>Assay to use in differential expression testing</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>data.frame with a ranked list of putative markers as rows, and associated
statistics as columns (p-values, ROC score, etc., depending on the test used (<code>test.use</code>)). The following columns are always present:</p><ul>
<li><p><code>avg_logFC</code>: log fold-chage of the average expression between the two groups. Positive values indicate that the gene is more highly expressed in the first group</p></li>
<li><p><code>pct.1</code>: The percentage of cells where the gene is detected in the first group</p></li>
<li><p><code>pct.2</code>: The percentage of cells where the gene is detected in the second group</p></li>
<li><p><code>p_val_adj</code>: Adjusted p-value, based on bonferroni correction using all genes in the dataset</p></li>
</ul>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>p-value adjustment is performed using bonferroni correction based on
the total number of genes in the dataset. Other correction methods are not
recommended, as Seurat pre-filters genes using the arguments above, reducing
the number of tests performed. Lastly, as Aaron Lun has pointed out, p-values
should be interpreted cautiously, as the genes used for clustering are the
same genes tested for differential expression.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>McDavid A, Finak G, Chattopadyay PK, et al. Data exploration,
quality control and testing in single-cell qPCR-based gene expression experiments.
Bioinformatics. 2013;29(4):461-467. doi:10.1093/bioinformatics/bts714</p>
<p>Trapnell C, et al. The dynamics and regulators of cell fate
decisions are revealed by pseudotemporal ordering of single cells. Nature
Biotechnology volume 32, pages 381-386 (2014)</p>
<p>Andrew McDavid, Greg Finak and Masanao Yajima (2017). MAST: Model-based
Analysis of Single Cell Transcriptomics. R package version 1.2.1.
https://github.com/RGLab/MAST/</p>
<p>Love MI, Huber W and Anders S (2014). "Moderated estimation of
fold change and dispersion for RNA-seq data with DESeq2." Genome Biology.
https://bioconductor.org/packages/release/bioc/html/DESeq2.html</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># Find markers for cluster 2</span>
<span class='no'>markers</span> <span class='kw'>&lt;-</span> <span class='fu'>FindMarkers</span>(<span class='kw'>object</span> <span class='kw'>=</span> <span class='no'>pbmc_small</span>, <span class='kw'>ident.1</span> <span class='kw'>=</span> <span class='fl'>2</span>)
<span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span>(<span class='kw'>x</span> <span class='kw'>=</span> <span class='no'>markers</span>)</div><div class='output co'>#&gt;                 p_val avg_logFC pct.1 pct.2    p_val_adj
#&gt; HLA-DPB1 2.667056e-08  1.749094 0.947 0.410 6.134228e-06
#&gt; MS4A1    7.532290e-08  4.533790 0.526 0.033 1.732427e-05
#&gt; HLA-DQB1 2.436852e-07  2.049598 0.789 0.246 5.604759e-05
#&gt; HLA-DRB1 1.341885e-06  1.659090 0.842 0.410 3.086335e-04
#&gt; HLA-DRA  4.618445e-06  1.812419 0.895 0.623 1.062242e-03
#&gt; TCL1A    6.943137e-06  4.079870 0.368 0.016 1.596922e-03</div><div class='input'>
<span class='co'># Take all cells in cluster 2, and find markers that separate cells in the 'g1' group (metadata</span>
<span class='co'># variable 'group')</span>
<span class='fu'><a href='https://rdrr.io/r/base/warning.html'>suppressWarnings</a></span>(<span class='no'>markers</span> <span class='kw'>&lt;-</span> <span class='fu'>FindMarkers</span>(<span class='no'>pbmc_small</span>, <span class='kw'>ident.1</span> <span class='kw'>=</span> <span class='st'>"g1"</span>, <span class='kw'>group.by</span> <span class='kw'>=</span> <span class='st'>'groups'</span>, <span class='kw'>subset.ident</span> <span class='kw'>=</span> <span class='st'>"2"</span>))
<span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span>(<span class='kw'>x</span> <span class='kw'>=</span> <span class='no'>markers</span>)</div><div class='output co'>#&gt;                p_val avg_logFC pct.1 pct.2 p_val_adj
#&gt; GSTP1     0.01601528  1.760034   0.7 0.111         1
#&gt; TPM4      0.02048683  3.037198   0.5 0.000         1
#&gt; LINC00936 0.02048683  2.835565   0.5 0.000         1
#&gt; IFI30     0.04515259  3.241233   0.4 0.000         1
#&gt; LGALS2    0.04515259  2.980942   0.4 0.000         1
#&gt; RHOC      0.04515259  2.726791   0.4 0.000         1</div><div class='input'>
# Pass 'clustertree' or an object of class phylo to ident.1 and
# a node to ident.2 as a replacement for FindMarkersNode
# pbmc_small <- BuildClusterTree(object = pbmc_small)
# markers <- FindMarkers(object = pbmc_small, ident.1 = 'clustertree', ident.2 = 5)
# head(x = markers)

</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
      <li><a href="#details">Details</a></li>
      <li><a href="#references">References</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by J. Sebastian Paez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


